// @chainlink/cre-sdk/dist/generated/networks.js
// This file is auto-generated. Do not edit manually.
// Generated from: https://github.com/smartcontractkit/chain-selectors
import mainnet_aptos_aptos_mainnet from './chain-selectors/mainnet/aptos/aptos-mainnet';
import mainnet_evm_0g_mainnet from './chain-selectors/mainnet/evm/0g-mainnet';
import mainnet_evm_abstract_mainnet from './chain-selectors/mainnet/evm/abstract-mainnet';
import mainnet_evm_apechain_mainnet from './chain-selectors/mainnet/evm/apechain-mainnet';
import mainnet_evm_areon_mainnet from './chain-selectors/mainnet/evm/areon-mainnet';
import mainnet_evm_avalanche_mainnet from './chain-selectors/mainnet/evm/avalanche-mainnet';
import mainnet_evm_avalanche_subnet_dexalot_mainnet from './chain-selectors/mainnet/evm/avalanche-subnet-dexalot-mainnet';
import mainnet_evm_berachain_mainnet from './chain-selectors/mainnet/evm/berachain-mainnet';
import mainnet_evm_binance_smart_chain_mainnet from './chain-selectors/mainnet/evm/binance.smart.chain-mainnet';
import mainnet_evm_binance_smart_chain_mainnet_opbnb_1 from './chain-selectors/mainnet/evm/binance.smart.chain-mainnet-opbnb-1';
import mainnet_evm_bitcichain_mainnet from './chain-selectors/mainnet/evm/bitcichain-mainnet';
import mainnet_evm_bitcoin_mainnet_bitlayer_1 from './chain-selectors/mainnet/evm/bitcoin-mainnet-bitlayer-1';
import mainnet_evm_bitcoin_mainnet_bob_1 from './chain-selectors/mainnet/evm/bitcoin-mainnet-bob-1';
import mainnet_evm_bitcoin_mainnet_botanix from './chain-selectors/mainnet/evm/bitcoin-mainnet-botanix';
import mainnet_evm_bitcoin_mainnet_bsquared_1 from './chain-selectors/mainnet/evm/bitcoin-mainnet-bsquared-1';
import mainnet_evm_bitcoin_merlin_mainnet from './chain-selectors/mainnet/evm/bitcoin-merlin-mainnet';
import mainnet_evm_bittensor_mainnet from './chain-selectors/mainnet/evm/bittensor-mainnet';
import mainnet_evm_bittorrent_chain_mainnet from './chain-selectors/mainnet/evm/bittorrent.chain-mainnet';
import mainnet_evm_celo_mainnet from './chain-selectors/mainnet/evm/celo-mainnet';
import mainnet_evm_codex_mainnet from './chain-selectors/mainnet/evm/codex-mainnet';
import mainnet_evm_coinex_smart_chain_mainnet from './chain-selectors/mainnet/evm/coinex.smart.chain-mainnet';
import mainnet_evm_conflux_mainnet from './chain-selectors/mainnet/evm/conflux-mainnet';
import mainnet_evm_core_mainnet from './chain-selectors/mainnet/evm/core-mainnet';
import mainnet_evm_corn_mainnet from './chain-selectors/mainnet/evm/corn-mainnet';
import mainnet_evm_cronos_mainnet from './chain-selectors/mainnet/evm/cronos-mainnet';
import mainnet_evm_cronos_zkevm_mainnet from './chain-selectors/mainnet/evm/cronos-zkevm-mainnet';
// Import all individual network files
import mainnet_evm_ethereum_mainnet from './chain-selectors/mainnet/evm/ethereum-mainnet';
import mainnet_evm_ethereum_mainnet_arbitrum_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-arbitrum-1';
import mainnet_evm_ethereum_mainnet_arbitrum_1_l3x_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-arbitrum-1-l3x-1';
import mainnet_evm_ethereum_mainnet_arbitrum_1_treasure_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-arbitrum-1-treasure-1';
import mainnet_evm_ethereum_mainnet_astar_zkevm_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-astar-zkevm-1';
import mainnet_evm_ethereum_mainnet_base_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-base-1';
import mainnet_evm_ethereum_mainnet_blast_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-blast-1';
import mainnet_evm_ethereum_mainnet_hashkey_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-hashkey-1';
import mainnet_evm_ethereum_mainnet_immutable_zkevm_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-immutable-zkevm-1';
import mainnet_evm_ethereum_mainnet_ink_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-ink-1';
import mainnet_evm_ethereum_mainnet_kroma_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-kroma-1';
import mainnet_evm_ethereum_mainnet_linea_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-linea-1';
import mainnet_evm_ethereum_mainnet_mantle_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-mantle-1';
import mainnet_evm_ethereum_mainnet_metis_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-metis-1';
import mainnet_evm_ethereum_mainnet_mode_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-mode-1';
import mainnet_evm_ethereum_mainnet_optimism_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-optimism-1';
import mainnet_evm_ethereum_mainnet_polygon_zkevm_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-polygon-zkevm-1';
import mainnet_evm_ethereum_mainnet_scroll_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-scroll-1';
import mainnet_evm_ethereum_mainnet_taiko_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-taiko-1';
import mainnet_evm_ethereum_mainnet_unichain_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-unichain-1';
import mainnet_evm_ethereum_mainnet_worldchain_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-worldchain-1';
import mainnet_evm_ethereum_mainnet_xlayer_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-xlayer-1';
import mainnet_evm_ethereum_mainnet_zircuit_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-zircuit-1';
import mainnet_evm_ethereum_mainnet_zksync_1 from './chain-selectors/mainnet/evm/ethereum-mainnet-zksync-1';
import mainnet_evm_etherlink_mainnet from './chain-selectors/mainnet/evm/etherlink-mainnet';
import mainnet_evm_fantom_mainnet from './chain-selectors/mainnet/evm/fantom-mainnet';
import mainnet_evm_filecoin_mainnet from './chain-selectors/mainnet/evm/filecoin-mainnet';
import mainnet_evm_fraxtal_mainnet from './chain-selectors/mainnet/evm/fraxtal-mainnet';
import mainnet_evm_gnosis_chain_mainnet from './chain-selectors/mainnet/evm/gnosis.chain-mainnet';
import mainnet_evm_hedera_mainnet from './chain-selectors/mainnet/evm/hedera-mainnet';
import mainnet_evm_hemi_mainnet from './chain-selectors/mainnet/evm/hemi-mainnet';
import mainnet_evm_hyperliquid_mainnet from './chain-selectors/mainnet/evm/hyperliquid-mainnet';
import mainnet_evm_janction_mainnet from './chain-selectors/mainnet/evm/janction-mainnet';
import mainnet_evm_kaia_mainnet from './chain-selectors/mainnet/evm/kaia-mainnet';
import mainnet_evm_kava_mainnet from './chain-selectors/mainnet/evm/kava-mainnet';
import mainnet_evm_kusama_mainnet_moonriver from './chain-selectors/mainnet/evm/kusama-mainnet-moonriver';
import mainnet_evm_lens_mainnet from './chain-selectors/mainnet/evm/lens-mainnet';
import mainnet_evm_lisk_mainnet from './chain-selectors/mainnet/evm/lisk-mainnet';
import mainnet_evm_memento_mainnet from './chain-selectors/mainnet/evm/memento-mainnet';
import mainnet_evm_metal_mainnet from './chain-selectors/mainnet/evm/metal-mainnet';
import mainnet_evm_mind_mainnet from './chain-selectors/mainnet/evm/mind-mainnet';
import mainnet_evm_mint_mainnet from './chain-selectors/mainnet/evm/mint-mainnet';
import mainnet_evm_monad_mainnet from './chain-selectors/mainnet/evm/monad-mainnet';
import mainnet_evm_morph_mainnet from './chain-selectors/mainnet/evm/morph-mainnet';
import mainnet_evm_near_mainnet from './chain-selectors/mainnet/evm/near-mainnet';
import mainnet_evm_neonlink_mainnet from './chain-selectors/mainnet/evm/neonlink-mainnet';
import mainnet_evm_neox_mainnet from './chain-selectors/mainnet/evm/neox-mainnet';
import mainnet_evm_nexon_mainnet_henesys from './chain-selectors/mainnet/evm/nexon-mainnet-henesys';
import mainnet_evm_nexon_mainnet_lith from './chain-selectors/mainnet/evm/nexon-mainnet-lith';
import mainnet_evm_nexon_qa from './chain-selectors/mainnet/evm/nexon-qa';
import mainnet_evm_nexon_stage from './chain-selectors/mainnet/evm/nexon-stage';
import mainnet_evm_nibiru_mainnet from './chain-selectors/mainnet/evm/nibiru-mainnet';
import mainnet_evm_plasma_mainnet from './chain-selectors/mainnet/evm/plasma-mainnet';
import mainnet_evm_plume_mainnet from './chain-selectors/mainnet/evm/plume-mainnet';
import mainnet_evm_polkadot_mainnet_astar from './chain-selectors/mainnet/evm/polkadot-mainnet-astar';
import mainnet_evm_polkadot_mainnet_centrifuge from './chain-selectors/mainnet/evm/polkadot-mainnet-centrifuge';
import mainnet_evm_polkadot_mainnet_darwinia from './chain-selectors/mainnet/evm/polkadot-mainnet-darwinia';
import mainnet_evm_polkadot_mainnet_moonbeam from './chain-selectors/mainnet/evm/polkadot-mainnet-moonbeam';
import mainnet_evm_polygon_mainnet from './chain-selectors/mainnet/evm/polygon-mainnet';
import mainnet_evm_polygon_mainnet_katana from './chain-selectors/mainnet/evm/polygon-mainnet-katana';
import mainnet_evm_ronin_mainnet from './chain-selectors/mainnet/evm/ronin-mainnet';
import mainnet_evm_rootstock_mainnet from './chain-selectors/mainnet/evm/rootstock-mainnet';
import mainnet_evm_sei_mainnet from './chain-selectors/mainnet/evm/sei-mainnet';
import mainnet_evm_shibarium_mainnet from './chain-selectors/mainnet/evm/shibarium-mainnet';
import mainnet_evm_soneium_mainnet from './chain-selectors/mainnet/evm/soneium-mainnet';
import mainnet_evm_sonic_mainnet from './chain-selectors/mainnet/evm/sonic-mainnet';
import mainnet_evm_superseed_mainnet from './chain-selectors/mainnet/evm/superseed-mainnet';
import mainnet_evm_tac_mainnet from './chain-selectors/mainnet/evm/tac-mainnet';
import mainnet_evm_telos_evm_mainnet from './chain-selectors/mainnet/evm/telos-evm-mainnet';
import mainnet_evm_treasure_mainnet from './chain-selectors/mainnet/evm/treasure-mainnet';
import mainnet_evm_tron_mainnet_evm from './chain-selectors/mainnet/evm/tron-mainnet-evm';
import mainnet_evm_velas_mainnet from './chain-selectors/mainnet/evm/velas-mainnet';
import mainnet_evm_wemix_mainnet from './chain-selectors/mainnet/evm/wemix-mainnet';
import mainnet_evm_xdc_mainnet from './chain-selectors/mainnet/evm/xdc-mainnet';
import mainnet_evm_zetachain_mainnet from './chain-selectors/mainnet/evm/zetachain-mainnet';
import mainnet_evm_zklink_nova_mainnet from './chain-selectors/mainnet/evm/zklink.nova-mainnet';
import mainnet_evm_zora_mainnet from './chain-selectors/mainnet/evm/zora-mainnet';
import mainnet_solana_solana_mainnet from './chain-selectors/mainnet/solana/solana-mainnet';
import mainnet_sui_sui_mainnet from './chain-selectors/mainnet/sui/sui-mainnet';
import mainnet_ton_ton_mainnet from './chain-selectors/mainnet/ton/ton-mainnet';
import mainnet_tron_tron_mainnet from './chain-selectors/mainnet/tron/tron-mainnet';
import testnet_aptos_aptos_localnet from './chain-selectors/testnet/aptos/aptos-localnet';
import testnet_aptos_aptos_testnet from './chain-selectors/testnet/aptos/aptos-testnet';
import testnet_evm_0g_testnet_galileo from './chain-selectors/testnet/evm/0g-testnet-galileo';
import testnet_evm_0g_testnet_newton from './chain-selectors/testnet/evm/0g-testnet-newton';
import testnet_evm_abstract_testnet from './chain-selectors/testnet/evm/abstract-testnet';
import testnet_evm_anvil_devnet from './chain-selectors/testnet/evm/anvil-devnet';
import testnet_evm_apechain_testnet_curtis from './chain-selectors/testnet/evm/apechain-testnet-curtis';
import testnet_evm_areon_testnet from './chain-selectors/testnet/evm/areon-testnet';
import testnet_evm_avalanche_subnet_dexalot_testnet from './chain-selectors/testnet/evm/avalanche-subnet-dexalot-testnet';
import testnet_evm_avalanche_testnet_fuji from './chain-selectors/testnet/evm/avalanche-testnet-fuji';
import testnet_evm_avalanche_testnet_nexon from './chain-selectors/testnet/evm/avalanche-testnet-nexon';
import testnet_evm_berachain_testnet_artio from './chain-selectors/testnet/evm/berachain-testnet-artio';
import testnet_evm_berachain_testnet_bartio from './chain-selectors/testnet/evm/berachain-testnet-bartio';
import testnet_evm_berachain_testnet_bepolia from './chain-selectors/testnet/evm/berachain-testnet-bepolia';
import testnet_evm_binance_smart_chain_testnet from './chain-selectors/testnet/evm/binance.smart.chain-testnet';
import testnet_evm_binance_smart_chain_testnet_opbnb_1 from './chain-selectors/testnet/evm/binance.smart.chain-testnet-opbnb-1';
import testnet_evm_bitcichain_testnet from './chain-selectors/testnet/evm/bitcichain-testnet';
import testnet_evm_bitcoin_testnet_bitlayer_1 from './chain-selectors/testnet/evm/bitcoin-testnet-bitlayer-1';
import testnet_evm_bitcoin_testnet_botanix from './chain-selectors/testnet/evm/bitcoin-testnet-botanix';
import testnet_evm_bitcoin_testnet_bsquared_1 from './chain-selectors/testnet/evm/bitcoin-testnet-bsquared-1';
import testnet_evm_bitcoin_testnet_merlin from './chain-selectors/testnet/evm/bitcoin-testnet-merlin';
import testnet_evm_bitcoin_testnet_rootstock from './chain-selectors/testnet/evm/bitcoin-testnet-rootstock';
import testnet_evm_bitcoin_testnet_sepolia_bob_1 from './chain-selectors/testnet/evm/bitcoin-testnet-sepolia-bob-1';
import testnet_evm_bittensor_testnet from './chain-selectors/testnet/evm/bittensor-testnet';
import testnet_evm_bittorrent_chain_testnet from './chain-selectors/testnet/evm/bittorrent.chain-testnet';
import testnet_evm_celo_testnet_alfajores from './chain-selectors/testnet/evm/celo-testnet-alfajores';
import testnet_evm_codex_testnet from './chain-selectors/testnet/evm/codex-testnet';
import testnet_evm_coinex_smart_chain_testnet from './chain-selectors/testnet/evm/coinex.smart.chain-testnet';
import testnet_evm_core_testnet from './chain-selectors/testnet/evm/core-testnet';
import testnet_evm_cronos_testnet from './chain-selectors/testnet/evm/cronos-testnet';
import testnet_evm_cronos_testnet_zkevm_1 from './chain-selectors/testnet/evm/cronos-testnet-zkevm-1';
import testnet_evm_cronos_zkevm_testnet_sepolia from './chain-selectors/testnet/evm/cronos-zkevm-testnet-sepolia';
import testnet_evm_dtcc_testnet_andesite from './chain-selectors/testnet/evm/dtcc-testnet-andesite';
import testnet_evm_ethereum_testnet_goerli_arbitrum_1 from './chain-selectors/testnet/evm/ethereum-testnet-goerli-arbitrum-1';
import testnet_evm_ethereum_testnet_goerli_base_1 from './chain-selectors/testnet/evm/ethereum-testnet-goerli-base-1';
import testnet_evm_ethereum_testnet_goerli_linea_1 from './chain-selectors/testnet/evm/ethereum-testnet-goerli-linea-1';
import testnet_evm_ethereum_testnet_goerli_mantle_1 from './chain-selectors/testnet/evm/ethereum-testnet-goerli-mantle-1';
import testnet_evm_ethereum_testnet_goerli_optimism_1 from './chain-selectors/testnet/evm/ethereum-testnet-goerli-optimism-1';
import testnet_evm_ethereum_testnet_goerli_polygon_zkevm_1 from './chain-selectors/testnet/evm/ethereum-testnet-goerli-polygon-zkevm-1';
import testnet_evm_ethereum_testnet_goerli_zksync_1 from './chain-selectors/testnet/evm/ethereum-testnet-goerli-zksync-1';
import testnet_evm_ethereum_testnet_holesky from './chain-selectors/testnet/evm/ethereum-testnet-holesky';
import testnet_evm_ethereum_testnet_holesky_fraxtal_1 from './chain-selectors/testnet/evm/ethereum-testnet-holesky-fraxtal-1';
import testnet_evm_ethereum_testnet_holesky_morph_1 from './chain-selectors/testnet/evm/ethereum-testnet-holesky-morph-1';
import testnet_evm_ethereum_testnet_holesky_taiko_1 from './chain-selectors/testnet/evm/ethereum-testnet-holesky-taiko-1';
import testnet_evm_ethereum_testnet_sepolia from './chain-selectors/testnet/evm/ethereum-testnet-sepolia';
import testnet_evm_ethereum_testnet_sepolia_arbitrum_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-arbitrum-1';
import testnet_evm_ethereum_testnet_sepolia_arbitrum_1_l3x_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-arbitrum-1-l3x-1';
import testnet_evm_ethereum_testnet_sepolia_arbitrum_1_treasure_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-arbitrum-1-treasure-1';
import testnet_evm_ethereum_testnet_sepolia_base_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-base-1';
import testnet_evm_ethereum_testnet_sepolia_blast_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-blast-1';
import testnet_evm_ethereum_testnet_sepolia_corn_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-corn-1';
import testnet_evm_ethereum_testnet_sepolia_hashkey_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-hashkey-1';
import testnet_evm_ethereum_testnet_sepolia_immutable_zkevm_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-immutable-zkevm-1';
import testnet_evm_ethereum_testnet_sepolia_kroma_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-kroma-1';
import testnet_evm_ethereum_testnet_sepolia_lens_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-lens-1';
import testnet_evm_ethereum_testnet_sepolia_linea_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-linea-1';
import testnet_evm_ethereum_testnet_sepolia_lisk_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-lisk-1';
import testnet_evm_ethereum_testnet_sepolia_mantle_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-mantle-1';
import testnet_evm_ethereum_testnet_sepolia_metis_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-metis-1';
import testnet_evm_ethereum_testnet_sepolia_mode_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-mode-1';
import testnet_evm_ethereum_testnet_sepolia_optimism_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-optimism-1';
import testnet_evm_ethereum_testnet_sepolia_polygon_validium_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-polygon-validium-1';
import testnet_evm_ethereum_testnet_sepolia_polygon_zkevm_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-polygon-zkevm-1';
import testnet_evm_ethereum_testnet_sepolia_scroll_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-scroll-1';
import testnet_evm_ethereum_testnet_sepolia_soneium_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-soneium-1';
import testnet_evm_ethereum_testnet_sepolia_unichain_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-unichain-1';
import testnet_evm_ethereum_testnet_sepolia_worldchain_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-worldchain-1';
import testnet_evm_ethereum_testnet_sepolia_xlayer_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-xlayer-1';
import testnet_evm_ethereum_testnet_sepolia_zircuit_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-zircuit-1';
import testnet_evm_ethereum_testnet_sepolia_zksync_1 from './chain-selectors/testnet/evm/ethereum-testnet-sepolia-zksync-1';
import testnet_evm_etherlink_testnet from './chain-selectors/testnet/evm/etherlink-testnet';
import testnet_evm_fantom_testnet from './chain-selectors/testnet/evm/fantom-testnet';
import testnet_evm_filecoin_testnet from './chain-selectors/testnet/evm/filecoin-testnet';
import testnet_evm_geth_testnet from './chain-selectors/testnet/evm/geth-testnet';
import testnet_evm_gnosis_chain_testnet_chiado from './chain-selectors/testnet/evm/gnosis.chain-testnet-chiado';
import testnet_evm_hedera_testnet from './chain-selectors/testnet/evm/hedera-testnet';
import testnet_evm_hemi_testnet_sepolia from './chain-selectors/testnet/evm/hemi-testnet-sepolia';
import testnet_evm_hyperliquid_testnet from './chain-selectors/testnet/evm/hyperliquid-testnet';
import testnet_evm_ink_testnet_sepolia from './chain-selectors/testnet/evm/ink-testnet-sepolia';
import testnet_evm_janction_testnet_sepolia from './chain-selectors/testnet/evm/janction-testnet-sepolia';
import testnet_evm_jovay_testnet from './chain-selectors/testnet/evm/jovay-testnet';
import testnet_evm_kaia_testnet_kairos from './chain-selectors/testnet/evm/kaia-testnet-kairos';
import testnet_evm_kava_testnet from './chain-selectors/testnet/evm/kava-testnet';
import testnet_evm_megaeth_testnet from './chain-selectors/testnet/evm/megaeth-testnet';
import testnet_evm_memento_testnet from './chain-selectors/testnet/evm/memento-testnet';
import testnet_evm_metal_testnet from './chain-selectors/testnet/evm/metal-testnet';
import testnet_evm_mind_testnet from './chain-selectors/testnet/evm/mind-testnet';
import testnet_evm_mint_testnet from './chain-selectors/testnet/evm/mint-testnet';
import testnet_evm_monad_testnet from './chain-selectors/testnet/evm/monad-testnet';
import testnet_evm_near_testnet from './chain-selectors/testnet/evm/near-testnet';
import testnet_evm_neonlink_testnet from './chain-selectors/testnet/evm/neonlink-testnet';
import testnet_evm_neox_testnet_t4 from './chain-selectors/testnet/evm/neox-testnet-t4';
import testnet_evm_nexon_dev from './chain-selectors/testnet/evm/nexon-dev';
import testnet_evm_nibiru_testnet from './chain-selectors/testnet/evm/nibiru-testnet';
import testnet_evm_ondo_testnet from './chain-selectors/testnet/evm/ondo-testnet';
import testnet_evm_pharos_testnet from './chain-selectors/testnet/evm/pharos-testnet';
import testnet_evm_plasma_testnet from './chain-selectors/testnet/evm/plasma-testnet';
import testnet_evm_plume_devnet from './chain-selectors/testnet/evm/plume-devnet';
import testnet_evm_plume_testnet from './chain-selectors/testnet/evm/plume-testnet';
import testnet_evm_plume_testnet_sepolia from './chain-selectors/testnet/evm/plume-testnet-sepolia';
import testnet_evm_polkadot_testnet_astar_shibuya from './chain-selectors/testnet/evm/polkadot-testnet-astar-shibuya';
import testnet_evm_polkadot_testnet_centrifuge_altair from './chain-selectors/testnet/evm/polkadot-testnet-centrifuge-altair';
import testnet_evm_polkadot_testnet_darwinia_pangoro from './chain-selectors/testnet/evm/polkadot-testnet-darwinia-pangoro';
import testnet_evm_polkadot_testnet_moonbeam_moonbase from './chain-selectors/testnet/evm/polkadot-testnet-moonbeam-moonbase';
import testnet_evm_polygon_testnet_amoy from './chain-selectors/testnet/evm/polygon-testnet-amoy';
import testnet_evm_polygon_testnet_mumbai from './chain-selectors/testnet/evm/polygon-testnet-mumbai';
import testnet_evm_polygon_testnet_tatara from './chain-selectors/testnet/evm/polygon-testnet-tatara';
import testnet_evm_private_testnet_andesite from './chain-selectors/testnet/evm/private-testnet-andesite';
import testnet_evm_private_testnet_granite from './chain-selectors/testnet/evm/private-testnet-granite';
import testnet_evm_private_testnet_mica from './chain-selectors/testnet/evm/private-testnet-mica';
import testnet_evm_private_testnet_obsidian from './chain-selectors/testnet/evm/private-testnet-obsidian';
import testnet_evm_private_testnet_opala from './chain-selectors/testnet/evm/private-testnet-opala';
import testnet_evm_ronin_testnet_saigon from './chain-selectors/testnet/evm/ronin-testnet-saigon';
import testnet_evm_sei_testnet_atlantic from './chain-selectors/testnet/evm/sei-testnet-atlantic';
import testnet_evm_shibarium_testnet_puppynet from './chain-selectors/testnet/evm/shibarium-testnet-puppynet';
import testnet_evm_sonic_testnet_blaze from './chain-selectors/testnet/evm/sonic-testnet-blaze';
import testnet_evm_story_testnet from './chain-selectors/testnet/evm/story-testnet';
import testnet_evm_superseed_testnet from './chain-selectors/testnet/evm/superseed-testnet';
import testnet_evm_tac_testnet from './chain-selectors/testnet/evm/tac-testnet';
import testnet_evm_telos_evm_testnet from './chain-selectors/testnet/evm/telos-evm-testnet';
import testnet_evm_treasure_testnet_topaz from './chain-selectors/testnet/evm/treasure-testnet-topaz';
import testnet_evm_tron_devnet_evm from './chain-selectors/testnet/evm/tron-devnet-evm';
import testnet_evm_tron_testnet_nile_evm from './chain-selectors/testnet/evm/tron-testnet-nile-evm';
import testnet_evm_tron_testnet_shasta_evm from './chain-selectors/testnet/evm/tron-testnet-shasta-evm';
import testnet_evm_velas_testnet from './chain-selectors/testnet/evm/velas-testnet';
import testnet_evm_wemix_testnet from './chain-selectors/testnet/evm/wemix-testnet';
import testnet_evm_xdc_testnet from './chain-selectors/testnet/evm/xdc-testnet';
import testnet_evm_zero_g_testnet_galileo from './chain-selectors/testnet/evm/zero-g-testnet-galileo';
import testnet_evm_zircuit_testnet_garfield from './chain-selectors/testnet/evm/zircuit-testnet-garfield';
import testnet_evm_zklink_nova_testnet from './chain-selectors/testnet/evm/zklink.nova-testnet';
import testnet_evm_zora_testnet from './chain-selectors/testnet/evm/zora-testnet';
import testnet_solana_solana_devnet from './chain-selectors/testnet/solana/solana-devnet';
import testnet_solana_solana_testnet from './chain-selectors/testnet/solana/solana-testnet';
import testnet_sui_sui_localnet from './chain-selectors/testnet/sui/sui-localnet';
import testnet_sui_sui_testnet from './chain-selectors/testnet/sui/sui-testnet';
import testnet_ton_ton_localnet from './chain-selectors/testnet/ton/ton-localnet';
import testnet_ton_ton_testnet from './chain-selectors/testnet/ton/ton-testnet';
import testnet_tron_tron_devnet from './chain-selectors/testnet/tron/tron-devnet';
import testnet_tron_tron_testnet_nile from './chain-selectors/testnet/tron/tron-testnet-nile';
import testnet_tron_tron_testnet_shasta from './chain-selectors/testnet/tron/tron-testnet-shasta';
export const allNetworks = [
    mainnet_evm_ethereum_mainnet,
    mainnet_evm_ethereum_mainnet_optimism_1,
    mainnet_evm_cronos_mainnet,
    mainnet_evm_rootstock_mainnet,
    testnet_evm_bitcoin_testnet_rootstock,
    mainnet_evm_telos_evm_mainnet,
    testnet_evm_telos_evm_testnet,
    testnet_evm_polkadot_testnet_darwinia_pangoro,
    mainnet_evm_polkadot_mainnet_darwinia,
    mainnet_evm_xdc_mainnet,
    testnet_evm_xdc_testnet,
    mainnet_evm_coinex_smart_chain_mainnet,
    testnet_evm_coinex_smart_chain_testnet,
    mainnet_evm_binance_smart_chain_mainnet,
    testnet_evm_polkadot_testnet_astar_shibuya,
    testnet_evm_binance_smart_chain_testnet,
    mainnet_evm_gnosis_chain_mainnet,
    mainnet_evm_velas_mainnet,
    mainnet_evm_shibarium_mainnet,
    testnet_evm_velas_testnet,
    mainnet_evm_ethereum_mainnet_unichain_1,
    testnet_evm_ethereum_testnet_sepolia_hashkey_1,
    mainnet_evm_polygon_mainnet,
    mainnet_evm_monad_mainnet,
    mainnet_evm_sonic_mainnet,
    testnet_evm_shibarium_testnet_puppynet,
    mainnet_evm_ethereum_mainnet_hashkey_1,
    mainnet_evm_mint_mainnet,
    testnet_evm_ethereum_testnet_sepolia_xlayer_1,
    mainnet_evm_ethereum_mainnet_xlayer_1,
    mainnet_evm_bittorrent_chain_mainnet,
    mainnet_evm_binance_smart_chain_mainnet_opbnb_1,
    mainnet_evm_bitcoin_mainnet_bsquared_1,
    mainnet_evm_mind_mainnet,
    mainnet_evm_lens_mainnet,
    mainnet_evm_tac_mainnet,
    testnet_evm_cronos_zkevm_testnet_sepolia,
    mainnet_evm_fantom_mainnet,
    mainnet_evm_fraxtal_mainnet,
    mainnet_evm_ethereum_mainnet_kroma_1,
    mainnet_evm_neonlink_mainnet,
    testnet_evm_ethereum_testnet_goerli_zksync_1,
    testnet_evm_cronos_testnet_zkevm_1,
    mainnet_evm_hedera_mainnet,
    testnet_evm_hedera_testnet,
    testnet_evm_ethereum_testnet_sepolia_zksync_1,
    mainnet_evm_filecoin_mainnet,
    mainnet_evm_ethereum_mainnet_zksync_1,
    testnet_evm_cronos_testnet,
    mainnet_evm_cronos_zkevm_mainnet,
    mainnet_evm_near_mainnet,
    testnet_evm_near_testnet,
    testnet_evm_ethereum_testnet_goerli_optimism_1,
    testnet_evm_areon_testnet,
    mainnet_evm_areon_mainnet,
    mainnet_evm_ethereum_mainnet_worldchain_1,
    mainnet_evm_polkadot_mainnet_astar,
    mainnet_evm_janction_mainnet,
    testnet_evm_janction_testnet_sepolia,
    testnet_evm_private_testnet_obsidian,
    testnet_evm_ethereum_testnet_sepolia_mode_1,
    testnet_evm_bittensor_testnet,
    mainnet_evm_bittensor_mainnet,
    testnet_evm_hyperliquid_testnet,
    mainnet_evm_hyperliquid_mainnet,
    testnet_evm_kaia_testnet_kairos,
    testnet_evm_bittorrent_chain_testnet,
    mainnet_evm_conflux_mainnet,
    mainnet_evm_ethereum_mainnet_metis_1,
    mainnet_evm_ethereum_mainnet_polygon_zkevm_1,
    mainnet_evm_wemix_mainnet,
    testnet_evm_wemix_testnet,
    testnet_evm_core_testnet,
    mainnet_evm_core_mainnet,
    testnet_evm_bitcoin_testnet_bsquared_1,
    mainnet_evm_lisk_mainnet,
    mainnet_evm_polkadot_mainnet_moonbeam,
    mainnet_evm_kusama_mainnet_moonriver,
    testnet_evm_polkadot_testnet_moonbeam_moonbase,
    testnet_evm_ethereum_testnet_sepolia_unichain_1,
    testnet_evm_sei_testnet_atlantic,
    mainnet_evm_sei_mainnet,
    testnet_evm_geth_testnet,
    testnet_evm_ethereum_testnet_goerli_polygon_zkevm_1,
    testnet_evm_story_testnet,
    testnet_evm_mint_testnet,
    testnet_evm_metal_testnet,
    mainnet_evm_metal_mainnet,
    mainnet_evm_soneium_mainnet,
    mainnet_evm_bitcichain_mainnet,
    testnet_evm_bitcichain_testnet,
    testnet_evm_ethereum_testnet_sepolia_soneium_1,
    mainnet_evm_ronin_mainnet,
    testnet_evm_ronin_testnet_saigon,
    testnet_evm_private_testnet_granite,
    testnet_evm_private_testnet_andesite,
    testnet_evm_dtcc_testnet_andesite,
    mainnet_evm_polkadot_mainnet_centrifuge,
    testnet_evm_polkadot_testnet_centrifuge_altair,
    testnet_evm_memento_testnet,
    testnet_evm_kava_testnet,
    mainnet_evm_kava_mainnet,
    testnet_evm_ethereum_testnet_sepolia_kroma_1,
    testnet_evm_tac_testnet,
    testnet_evm_ethereum_testnet_sepolia_polygon_zkevm_1,
    testnet_evm_ethereum_testnet_holesky_fraxtal_1,
    mainnet_evm_abstract_mainnet,
    testnet_evm_ethereum_testnet_holesky_morph_1,
    mainnet_evm_morph_mainnet,
    testnet_evm_bitcoin_testnet_botanix,
    mainnet_evm_bitcoin_mainnet_botanix,
    mainnet_evm_ethereum_mainnet_astar_zkevm_1,
    testnet_evm_fantom_testnet,
    mainnet_evm_bitcoin_merlin_mainnet,
    testnet_evm_ethereum_testnet_sepolia_lisk_1,
    testnet_evm_ethereum_testnet_sepolia_worldchain_1,
    mainnet_evm_ethereum_mainnet_mantle_1,
    testnet_evm_ethereum_testnet_goerli_mantle_1,
    testnet_evm_ethereum_testnet_sepolia_mantle_1,
    mainnet_evm_superseed_mainnet,
    testnet_evm_binance_smart_chain_testnet_opbnb_1,
    testnet_evm_nexon_dev,
    testnet_evm_megaeth_testnet,
    mainnet_evm_nibiru_mainnet,
    testnet_evm_nibiru_testnet,
    mainnet_evm_zetachain_mainnet,
    mainnet_evm_kaia_mainnet,
    mainnet_evm_ethereum_mainnet_base_1,
    testnet_evm_ondo_testnet,
    testnet_evm_neonlink_testnet,
    mainnet_evm_plasma_mainnet,
    testnet_evm_plasma_testnet,
    testnet_evm_monad_testnet,
    testnet_evm_gnosis_chain_testnet_chiado,
    testnet_evm_abstract_testnet,
    mainnet_evm_ethereum_mainnet_arbitrum_1_l3x_1,
    testnet_evm_ethereum_testnet_sepolia_arbitrum_1_l3x_1,
    mainnet_evm_ethereum_mainnet_immutable_zkevm_1,
    testnet_evm_ethereum_testnet_sepolia_immutable_zkevm_1,
    testnet_evm_0g_testnet_newton,
    testnet_evm_0g_testnet_galileo,
    mainnet_evm_0g_mainnet,
    testnet_evm_ethereum_testnet_holesky,
    testnet_evm_anvil_devnet,
    testnet_evm_apechain_testnet_curtis,
    mainnet_evm_apechain_mainnet,
    mainnet_evm_ethereum_mainnet_mode_1,
    testnet_evm_ethereum_testnet_sepolia_lens_1,
    mainnet_evm_ethereum_mainnet_arbitrum_1,
    mainnet_evm_celo_mainnet,
    mainnet_evm_etherlink_mainnet,
    mainnet_evm_hemi_mainnet,
    testnet_evm_avalanche_testnet_fuji,
    mainnet_evm_avalanche_mainnet,
    testnet_evm_celo_testnet_alfajores,
    testnet_evm_private_testnet_opala,
    mainnet_evm_neox_mainnet,
    testnet_evm_zircuit_testnet_garfield,
    testnet_evm_ethereum_testnet_sepolia_zircuit_1,
    mainnet_evm_ethereum_mainnet_zircuit_1,
    mainnet_evm_memento_mainnet,
    testnet_evm_superseed_testnet,
    testnet_evm_sonic_testnet_blaze,
    mainnet_evm_ethereum_mainnet_ink_1,
    testnet_evm_ethereum_testnet_goerli_linea_1,
    testnet_evm_ethereum_testnet_sepolia_linea_1,
    mainnet_evm_ethereum_mainnet_linea_1,
    testnet_evm_ethereum_testnet_sepolia_metis_1,
    mainnet_evm_nexon_mainnet_lith,
    mainnet_evm_bitcoin_mainnet_bob_1,
    mainnet_evm_treasure_mainnet,
    mainnet_evm_nexon_mainnet_henesys,
    testnet_evm_polygon_testnet_mumbai,
    testnet_evm_polygon_testnet_amoy,
    testnet_evm_berachain_testnet_bepolia,
    testnet_evm_berachain_testnet_bartio,
    testnet_evm_berachain_testnet_artio,
    testnet_evm_zero_g_testnet_galileo,
    mainnet_evm_berachain_mainnet,
    mainnet_evm_codex_mainnet,
    mainnet_evm_ethereum_mainnet_blast_1,
    testnet_evm_ethereum_testnet_goerli_base_1,
    testnet_evm_ethereum_testnet_sepolia_base_1,
    testnet_evm_plume_devnet,
    mainnet_evm_plume_mainnet,
    testnet_evm_plume_testnet_sepolia,
    testnet_evm_etherlink_testnet,
    testnet_evm_polygon_testnet_tatara,
    mainnet_evm_ethereum_mainnet_taiko_1,
    testnet_evm_ethereum_testnet_holesky_taiko_1,
    testnet_evm_mind_testnet,
    testnet_evm_bitcoin_testnet_bitlayer_1,
    mainnet_evm_bitcoin_mainnet_bitlayer_1,
    testnet_evm_ethereum_testnet_goerli_arbitrum_1,
    testnet_evm_ethereum_testnet_sepolia_arbitrum_1,
    testnet_evm_private_testnet_mica,
    testnet_evm_avalanche_subnet_dexalot_testnet,
    mainnet_evm_avalanche_subnet_dexalot_mainnet,
    testnet_evm_ethereum_testnet_sepolia_scroll_1,
    mainnet_evm_ethereum_mainnet_scroll_1,
    testnet_evm_avalanche_testnet_nexon,
    testnet_evm_bitcoin_testnet_merlin,
    testnet_evm_pharos_testnet,
    testnet_evm_ethereum_testnet_sepolia_polygon_validium_1,
    testnet_evm_hemi_testnet_sepolia,
    mainnet_evm_polygon_mainnet_katana,
    testnet_evm_ink_testnet_sepolia,
    mainnet_evm_nexon_qa,
    testnet_evm_bitcoin_testnet_sepolia_bob_1,
    mainnet_evm_zklink_nova_mainnet,
    testnet_evm_zklink_nova_testnet,
    testnet_evm_codex_testnet,
    mainnet_evm_nexon_stage,
    testnet_evm_ethereum_testnet_sepolia_arbitrum_1_treasure_1,
    testnet_evm_treasure_testnet_topaz,
    mainnet_evm_ethereum_mainnet_arbitrum_1_treasure_1,
    testnet_evm_jovay_testnet,
    mainnet_evm_zora_mainnet,
    testnet_evm_ethereum_testnet_sepolia,
    testnet_evm_ethereum_testnet_sepolia_optimism_1,
    testnet_evm_neox_testnet_t4,
    mainnet_evm_corn_mainnet,
    testnet_evm_ethereum_testnet_sepolia_corn_1,
    testnet_evm_filecoin_testnet,
    testnet_evm_plume_testnet,
    testnet_evm_ethereum_testnet_sepolia_blast_1,
    mainnet_evm_tron_mainnet_evm,
    testnet_evm_zora_testnet,
    testnet_evm_tron_testnet_shasta_evm,
    testnet_evm_tron_devnet_evm,
    testnet_evm_tron_testnet_nile_evm,
    mainnet_solana_solana_mainnet,
    testnet_solana_solana_testnet,
    testnet_solana_solana_devnet,
    mainnet_aptos_aptos_mainnet,
    testnet_aptos_aptos_testnet,
    testnet_aptos_aptos_localnet,
    mainnet_sui_sui_mainnet,
    testnet_sui_sui_testnet,
    testnet_sui_sui_localnet,
    mainnet_ton_ton_mainnet,
    testnet_ton_ton_testnet,
    testnet_ton_ton_localnet,
    mainnet_tron_tron_mainnet,
    testnet_tron_tron_testnet_shasta,
    testnet_tron_tron_devnet,
    testnet_tron_tron_testnet_nile,
];
export const mainnet = {
    evm: [
        mainnet_evm_ethereum_mainnet,
        mainnet_evm_ethereum_mainnet_optimism_1,
        mainnet_evm_cronos_mainnet,
        mainnet_evm_rootstock_mainnet,
        mainnet_evm_telos_evm_mainnet,
        mainnet_evm_polkadot_mainnet_darwinia,
        mainnet_evm_xdc_mainnet,
        mainnet_evm_coinex_smart_chain_mainnet,
        mainnet_evm_binance_smart_chain_mainnet,
        mainnet_evm_gnosis_chain_mainnet,
        mainnet_evm_velas_mainnet,
        mainnet_evm_shibarium_mainnet,
        mainnet_evm_ethereum_mainnet_unichain_1,
        mainnet_evm_polygon_mainnet,
        mainnet_evm_monad_mainnet,
        mainnet_evm_sonic_mainnet,
        mainnet_evm_ethereum_mainnet_hashkey_1,
        mainnet_evm_mint_mainnet,
        mainnet_evm_ethereum_mainnet_xlayer_1,
        mainnet_evm_bittorrent_chain_mainnet,
        mainnet_evm_binance_smart_chain_mainnet_opbnb_1,
        mainnet_evm_bitcoin_mainnet_bsquared_1,
        mainnet_evm_mind_mainnet,
        mainnet_evm_lens_mainnet,
        mainnet_evm_tac_mainnet,
        mainnet_evm_fantom_mainnet,
        mainnet_evm_fraxtal_mainnet,
        mainnet_evm_ethereum_mainnet_kroma_1,
        mainnet_evm_neonlink_mainnet,
        mainnet_evm_hedera_mainnet,
        mainnet_evm_filecoin_mainnet,
        mainnet_evm_ethereum_mainnet_zksync_1,
        mainnet_evm_cronos_zkevm_mainnet,
        mainnet_evm_near_mainnet,
        mainnet_evm_areon_mainnet,
        mainnet_evm_ethereum_mainnet_worldchain_1,
        mainnet_evm_polkadot_mainnet_astar,
        mainnet_evm_janction_mainnet,
        mainnet_evm_bittensor_mainnet,
        mainnet_evm_hyperliquid_mainnet,
        mainnet_evm_conflux_mainnet,
        mainnet_evm_ethereum_mainnet_metis_1,
        mainnet_evm_ethereum_mainnet_polygon_zkevm_1,
        mainnet_evm_wemix_mainnet,
        mainnet_evm_core_mainnet,
        mainnet_evm_lisk_mainnet,
        mainnet_evm_polkadot_mainnet_moonbeam,
        mainnet_evm_kusama_mainnet_moonriver,
        mainnet_evm_sei_mainnet,
        mainnet_evm_metal_mainnet,
        mainnet_evm_soneium_mainnet,
        mainnet_evm_bitcichain_mainnet,
        mainnet_evm_ronin_mainnet,
        mainnet_evm_polkadot_mainnet_centrifuge,
        mainnet_evm_kava_mainnet,
        mainnet_evm_abstract_mainnet,
        mainnet_evm_morph_mainnet,
        mainnet_evm_bitcoin_mainnet_botanix,
        mainnet_evm_ethereum_mainnet_astar_zkevm_1,
        mainnet_evm_bitcoin_merlin_mainnet,
        mainnet_evm_ethereum_mainnet_mantle_1,
        mainnet_evm_superseed_mainnet,
        mainnet_evm_nibiru_mainnet,
        mainnet_evm_zetachain_mainnet,
        mainnet_evm_kaia_mainnet,
        mainnet_evm_ethereum_mainnet_base_1,
        mainnet_evm_plasma_mainnet,
        mainnet_evm_ethereum_mainnet_arbitrum_1_l3x_1,
        mainnet_evm_ethereum_mainnet_immutable_zkevm_1,
        mainnet_evm_0g_mainnet,
        mainnet_evm_apechain_mainnet,
        mainnet_evm_ethereum_mainnet_mode_1,
        mainnet_evm_ethereum_mainnet_arbitrum_1,
        mainnet_evm_celo_mainnet,
        mainnet_evm_etherlink_mainnet,
        mainnet_evm_hemi_mainnet,
        mainnet_evm_avalanche_mainnet,
        mainnet_evm_neox_mainnet,
        mainnet_evm_ethereum_mainnet_zircuit_1,
        mainnet_evm_memento_mainnet,
        mainnet_evm_ethereum_mainnet_ink_1,
        mainnet_evm_ethereum_mainnet_linea_1,
        mainnet_evm_nexon_mainnet_lith,
        mainnet_evm_bitcoin_mainnet_bob_1,
        mainnet_evm_treasure_mainnet,
        mainnet_evm_nexon_mainnet_henesys,
        mainnet_evm_berachain_mainnet,
        mainnet_evm_codex_mainnet,
        mainnet_evm_ethereum_mainnet_blast_1,
        mainnet_evm_plume_mainnet,
        mainnet_evm_ethereum_mainnet_taiko_1,
        mainnet_evm_bitcoin_mainnet_bitlayer_1,
        mainnet_evm_avalanche_subnet_dexalot_mainnet,
        mainnet_evm_ethereum_mainnet_scroll_1,
        mainnet_evm_polygon_mainnet_katana,
        mainnet_evm_nexon_qa,
        mainnet_evm_zklink_nova_mainnet,
        mainnet_evm_nexon_stage,
        mainnet_evm_ethereum_mainnet_arbitrum_1_treasure_1,
        mainnet_evm_zora_mainnet,
        mainnet_evm_corn_mainnet,
        mainnet_evm_tron_mainnet_evm,
    ],
    solana: [mainnet_solana_solana_mainnet],
    aptos: [mainnet_aptos_aptos_mainnet],
    sui: [mainnet_sui_sui_mainnet],
    ton: [mainnet_ton_ton_mainnet],
    tron: [mainnet_tron_tron_mainnet],
};
export const testnet = {
    evm: [
        testnet_evm_bitcoin_testnet_rootstock,
        testnet_evm_telos_evm_testnet,
        testnet_evm_polkadot_testnet_darwinia_pangoro,
        testnet_evm_xdc_testnet,
        testnet_evm_coinex_smart_chain_testnet,
        testnet_evm_polkadot_testnet_astar_shibuya,
        testnet_evm_binance_smart_chain_testnet,
        testnet_evm_velas_testnet,
        testnet_evm_ethereum_testnet_sepolia_hashkey_1,
        testnet_evm_shibarium_testnet_puppynet,
        testnet_evm_ethereum_testnet_sepolia_xlayer_1,
        testnet_evm_cronos_zkevm_testnet_sepolia,
        testnet_evm_ethereum_testnet_goerli_zksync_1,
        testnet_evm_cronos_testnet_zkevm_1,
        testnet_evm_hedera_testnet,
        testnet_evm_ethereum_testnet_sepolia_zksync_1,
        testnet_evm_cronos_testnet,
        testnet_evm_near_testnet,
        testnet_evm_ethereum_testnet_goerli_optimism_1,
        testnet_evm_areon_testnet,
        testnet_evm_janction_testnet_sepolia,
        testnet_evm_private_testnet_obsidian,
        testnet_evm_ethereum_testnet_sepolia_mode_1,
        testnet_evm_bittensor_testnet,
        testnet_evm_hyperliquid_testnet,
        testnet_evm_kaia_testnet_kairos,
        testnet_evm_bittorrent_chain_testnet,
        testnet_evm_wemix_testnet,
        testnet_evm_core_testnet,
        testnet_evm_bitcoin_testnet_bsquared_1,
        testnet_evm_polkadot_testnet_moonbeam_moonbase,
        testnet_evm_ethereum_testnet_sepolia_unichain_1,
        testnet_evm_sei_testnet_atlantic,
        testnet_evm_geth_testnet,
        testnet_evm_ethereum_testnet_goerli_polygon_zkevm_1,
        testnet_evm_story_testnet,
        testnet_evm_mint_testnet,
        testnet_evm_metal_testnet,
        testnet_evm_bitcichain_testnet,
        testnet_evm_ethereum_testnet_sepolia_soneium_1,
        testnet_evm_ronin_testnet_saigon,
        testnet_evm_private_testnet_granite,
        testnet_evm_private_testnet_andesite,
        testnet_evm_dtcc_testnet_andesite,
        testnet_evm_polkadot_testnet_centrifuge_altair,
        testnet_evm_memento_testnet,
        testnet_evm_kava_testnet,
        testnet_evm_ethereum_testnet_sepolia_kroma_1,
        testnet_evm_tac_testnet,
        testnet_evm_ethereum_testnet_sepolia_polygon_zkevm_1,
        testnet_evm_ethereum_testnet_holesky_fraxtal_1,
        testnet_evm_ethereum_testnet_holesky_morph_1,
        testnet_evm_bitcoin_testnet_botanix,
        testnet_evm_fantom_testnet,
        testnet_evm_ethereum_testnet_sepolia_lisk_1,
        testnet_evm_ethereum_testnet_sepolia_worldchain_1,
        testnet_evm_ethereum_testnet_goerli_mantle_1,
        testnet_evm_ethereum_testnet_sepolia_mantle_1,
        testnet_evm_binance_smart_chain_testnet_opbnb_1,
        testnet_evm_nexon_dev,
        testnet_evm_megaeth_testnet,
        testnet_evm_nibiru_testnet,
        testnet_evm_ondo_testnet,
        testnet_evm_neonlink_testnet,
        testnet_evm_plasma_testnet,
        testnet_evm_monad_testnet,
        testnet_evm_gnosis_chain_testnet_chiado,
        testnet_evm_abstract_testnet,
        testnet_evm_ethereum_testnet_sepolia_arbitrum_1_l3x_1,
        testnet_evm_ethereum_testnet_sepolia_immutable_zkevm_1,
        testnet_evm_0g_testnet_newton,
        testnet_evm_0g_testnet_galileo,
        testnet_evm_ethereum_testnet_holesky,
        testnet_evm_anvil_devnet,
        testnet_evm_apechain_testnet_curtis,
        testnet_evm_ethereum_testnet_sepolia_lens_1,
        testnet_evm_avalanche_testnet_fuji,
        testnet_evm_celo_testnet_alfajores,
        testnet_evm_private_testnet_opala,
        testnet_evm_zircuit_testnet_garfield,
        testnet_evm_ethereum_testnet_sepolia_zircuit_1,
        testnet_evm_superseed_testnet,
        testnet_evm_sonic_testnet_blaze,
        testnet_evm_ethereum_testnet_goerli_linea_1,
        testnet_evm_ethereum_testnet_sepolia_linea_1,
        testnet_evm_ethereum_testnet_sepolia_metis_1,
        testnet_evm_polygon_testnet_mumbai,
        testnet_evm_polygon_testnet_amoy,
        testnet_evm_berachain_testnet_bepolia,
        testnet_evm_berachain_testnet_bartio,
        testnet_evm_berachain_testnet_artio,
        testnet_evm_zero_g_testnet_galileo,
        testnet_evm_ethereum_testnet_goerli_base_1,
        testnet_evm_ethereum_testnet_sepolia_base_1,
        testnet_evm_plume_devnet,
        testnet_evm_plume_testnet_sepolia,
        testnet_evm_etherlink_testnet,
        testnet_evm_polygon_testnet_tatara,
        testnet_evm_ethereum_testnet_holesky_taiko_1,
        testnet_evm_mind_testnet,
        testnet_evm_bitcoin_testnet_bitlayer_1,
        testnet_evm_ethereum_testnet_goerli_arbitrum_1,
        testnet_evm_ethereum_testnet_sepolia_arbitrum_1,
        testnet_evm_private_testnet_mica,
        testnet_evm_avalanche_subnet_dexalot_testnet,
        testnet_evm_ethereum_testnet_sepolia_scroll_1,
        testnet_evm_avalanche_testnet_nexon,
        testnet_evm_bitcoin_testnet_merlin,
        testnet_evm_pharos_testnet,
        testnet_evm_ethereum_testnet_sepolia_polygon_validium_1,
        testnet_evm_hemi_testnet_sepolia,
        testnet_evm_ink_testnet_sepolia,
        testnet_evm_bitcoin_testnet_sepolia_bob_1,
        testnet_evm_zklink_nova_testnet,
        testnet_evm_codex_testnet,
        testnet_evm_ethereum_testnet_sepolia_arbitrum_1_treasure_1,
        testnet_evm_treasure_testnet_topaz,
        testnet_evm_jovay_testnet,
        testnet_evm_ethereum_testnet_sepolia,
        testnet_evm_ethereum_testnet_sepolia_optimism_1,
        testnet_evm_neox_testnet_t4,
        testnet_evm_ethereum_testnet_sepolia_corn_1,
        testnet_evm_filecoin_testnet,
        testnet_evm_plume_testnet,
        testnet_evm_ethereum_testnet_sepolia_blast_1,
        testnet_evm_zora_testnet,
        testnet_evm_tron_testnet_shasta_evm,
        testnet_evm_tron_devnet_evm,
        testnet_evm_tron_testnet_nile_evm,
    ],
    solana: [testnet_solana_solana_testnet, testnet_solana_solana_devnet],
    aptos: [testnet_aptos_aptos_testnet, testnet_aptos_aptos_localnet],
    sui: [testnet_sui_sui_testnet, testnet_sui_sui_localnet],
    ton: [testnet_ton_ton_testnet, testnet_ton_ton_localnet],
    tron: [
        testnet_tron_tron_testnet_shasta,
        testnet_tron_tron_devnet,
        testnet_tron_tron_testnet_nile,
    ],
};
// Optimized Maps for fast lookups by chain selector (bigint keys)
export const mainnetBySelector = new Map([
    [5009297550715157269n, mainnet_evm_ethereum_mainnet],
    [3734403246176062136n, mainnet_evm_ethereum_mainnet_optimism_1],
    [1456215246176062136n, mainnet_evm_cronos_mainnet],
    [11964252391146578476n, mainnet_evm_rootstock_mainnet],
    [1477345371608778000n, mainnet_evm_telos_evm_mainnet],
    [8866418665544333000n, mainnet_evm_polkadot_mainnet_darwinia],
    [17673274061779414707n, mainnet_evm_xdc_mainnet],
    [1761333065194157300n, mainnet_evm_coinex_smart_chain_mainnet],
    [11344663589394136015n, mainnet_evm_binance_smart_chain_mainnet],
    [465200170687744372n, mainnet_evm_gnosis_chain_mainnet],
    [374210358663784372n, mainnet_evm_velas_mainnet],
    [3993510008929295315n, mainnet_evm_shibarium_mainnet],
    [1923510103922296319n, mainnet_evm_ethereum_mainnet_unichain_1],
    [4051577828743386545n, mainnet_evm_polygon_mainnet],
    [8481857512324358265n, mainnet_evm_monad_mainnet],
    [1673871237479749969n, mainnet_evm_sonic_mainnet],
    [7613811247471741961n, mainnet_evm_ethereum_mainnet_hashkey_1],
    [17164792800244661392n, mainnet_evm_mint_mainnet],
    [3016212468291539606n, mainnet_evm_ethereum_mainnet_xlayer_1],
    [3776006016387883143n, mainnet_evm_bittorrent_chain_mainnet],
    [465944652040885897n, mainnet_evm_binance_smart_chain_mainnet_opbnb_1],
    [5406759801798337480n, mainnet_evm_bitcoin_mainnet_bsquared_1],
    [11690709103138290329n, mainnet_evm_mind_mainnet],
    [5608378062013572713n, mainnet_evm_lens_mainnet],
    [5936861837188149645n, mainnet_evm_tac_mainnet],
    [3768048213127883732n, mainnet_evm_fantom_mainnet],
    [1462016016387883143n, mainnet_evm_fraxtal_mainnet],
    [3719320017875267166n, mainnet_evm_ethereum_mainnet_kroma_1],
    [8239338020728974000n, mainnet_evm_neonlink_mainnet],
    [3229138320728879060n, mainnet_evm_hedera_mainnet],
    [4561443241176882990n, mainnet_evm_filecoin_mainnet],
    [1562403441176082196n, mainnet_evm_ethereum_mainnet_zksync_1],
    [8788096068760390840n, mainnet_evm_cronos_zkevm_mainnet],
    [2039744413822257700n, mainnet_evm_near_mainnet],
    [1939936305787790600n, mainnet_evm_areon_mainnet],
    [2049429975587534727n, mainnet_evm_ethereum_mainnet_worldchain_1],
    [6422105447186081193n, mainnet_evm_polkadot_mainnet_astar],
    [9107126442626377432n, mainnet_evm_janction_mainnet],
    [2135107236357186872n, mainnet_evm_bittensor_mainnet],
    [2442541497099098535n, mainnet_evm_hyperliquid_mainnet],
    [3358365939762719202n, mainnet_evm_conflux_mainnet],
    [8805746078405598895n, mainnet_evm_ethereum_mainnet_metis_1],
    [4348158687435793198n, mainnet_evm_ethereum_mainnet_polygon_zkevm_1],
    [5142893604156789321n, mainnet_evm_wemix_mainnet],
    [1224752112135636129n, mainnet_evm_core_mainnet],
    [15293031020466096408n, mainnet_evm_lisk_mainnet],
    [1252863800116739621n, mainnet_evm_polkadot_mainnet_moonbeam],
    [1355020143337428062n, mainnet_evm_kusama_mainnet_moonriver],
    [9027416829622342829n, mainnet_evm_sei_mainnet],
    [13447077090413146373n, mainnet_evm_metal_mainnet],
    [12505351618335765396n, mainnet_evm_soneium_mainnet],
    [4874388048629246000n, mainnet_evm_bitcichain_mainnet],
    [6916147374840168594n, mainnet_evm_ronin_mainnet],
    [8175830712062617656n, mainnet_evm_polkadot_mainnet_centrifuge],
    [7550000543357438061n, mainnet_evm_kava_mainnet],
    [3577778157919314504n, mainnet_evm_abstract_mainnet],
    [18164309074156128038n, mainnet_evm_morph_mainnet],
    [4560701533377838164n, mainnet_evm_bitcoin_mainnet_botanix],
    [1540201334317828111n, mainnet_evm_ethereum_mainnet_astar_zkevm_1],
    [241851231317828981n, mainnet_evm_bitcoin_merlin_mainnet],
    [1556008542357238666n, mainnet_evm_ethereum_mainnet_mantle_1],
    [470401360549526817n, mainnet_evm_superseed_mainnet],
    [17349189558768828726n, mainnet_evm_nibiru_mainnet],
    [10817664450262215148n, mainnet_evm_zetachain_mainnet],
    [9813823125703490621n, mainnet_evm_kaia_mainnet],
    [15971525489660198786n, mainnet_evm_ethereum_mainnet_base_1],
    [9335212494177455608n, mainnet_evm_plasma_mainnet],
    [3162193654116181371n, mainnet_evm_ethereum_mainnet_arbitrum_1_l3x_1],
    [1237925231416731909n, mainnet_evm_ethereum_mainnet_immutable_zkevm_1],
    [4426351306075016396n, mainnet_evm_0g_mainnet],
    [14894068710063348487n, mainnet_evm_apechain_mainnet],
    [7264351850409363825n, mainnet_evm_ethereum_mainnet_mode_1],
    [4949039107694359620n, mainnet_evm_ethereum_mainnet_arbitrum_1],
    [1346049177634351622n, mainnet_evm_celo_mainnet],
    [13624601974233774587n, mainnet_evm_etherlink_mainnet],
    [1804312132722180201n, mainnet_evm_hemi_mainnet],
    [6433500567565415381n, mainnet_evm_avalanche_mainnet],
    [7222032299962346917n, mainnet_evm_neox_mainnet],
    [17198166215261833993n, mainnet_evm_ethereum_mainnet_zircuit_1],
    [6473245816409426016n, mainnet_evm_memento_mainnet],
    [3461204551265785888n, mainnet_evm_ethereum_mainnet_ink_1],
    [4627098889531055414n, mainnet_evm_ethereum_mainnet_linea_1],
    [15758750456714168963n, mainnet_evm_nexon_mainnet_lith],
    [3849287863852499584n, mainnet_evm_bitcoin_mainnet_bob_1],
    [5214452172935136222n, mainnet_evm_treasure_mainnet],
    [12657445206920369324n, mainnet_evm_nexon_mainnet_henesys],
    [1294465214383781161n, mainnet_evm_berachain_mainnet],
    [9478124434908827753n, mainnet_evm_codex_mainnet],
    [4411394078118774322n, mainnet_evm_ethereum_mainnet_blast_1],
    [17912061998839310979n, mainnet_evm_plume_mainnet],
    [16468599424800719238n, mainnet_evm_ethereum_mainnet_taiko_1],
    [7937294810946806131n, mainnet_evm_bitcoin_mainnet_bitlayer_1],
    [5463201557265485081n, mainnet_evm_avalanche_subnet_dexalot_mainnet],
    [13204309965629103672n, mainnet_evm_ethereum_mainnet_scroll_1],
    [2459028469735686113n, mainnet_evm_polygon_mainnet_katana],
    [14632960069656270105n, mainnet_evm_nexon_qa],
    [4350319965322101699n, mainnet_evm_zklink_nova_mainnet],
    [5556806327594153475n, mainnet_evm_nexon_stage],
    [1010349088906777999n, mainnet_evm_ethereum_mainnet_arbitrum_1_treasure_1],
    [3555797439612589184n, mainnet_evm_zora_mainnet],
    [9043146809313071210n, mainnet_evm_corn_mainnet],
    [1546563616611573946n, mainnet_evm_tron_mainnet_evm],
    [124615329519749607n, mainnet_solana_solana_mainnet],
    [4741433654826277614n, mainnet_aptos_aptos_mainnet],
    [17529533435026248318n, mainnet_sui_sui_mainnet],
    [16448340667252469081n, mainnet_ton_ton_mainnet],
    [1546563616611573945n, mainnet_tron_tron_mainnet],
]);
export const testnetBySelector = new Map([
    [8953668971247136127n, testnet_evm_bitcoin_testnet_rootstock],
    [729797994450396300n, testnet_evm_telos_evm_testnet],
    [4340886533089894000n, testnet_evm_polkadot_testnet_darwinia_pangoro],
    [3017758115101368649n, testnet_evm_xdc_testnet],
    [8955032871639343000n, testnet_evm_coinex_smart_chain_testnet],
    [6955638871347136141n, testnet_evm_polkadot_testnet_astar_shibuya],
    [13264668187771770619n, testnet_evm_binance_smart_chain_testnet],
    [572210378683744374n, testnet_evm_velas_testnet],
    [4356164186791070119n, testnet_evm_ethereum_testnet_sepolia_hashkey_1],
    [17833296867764334567n, testnet_evm_shibarium_testnet_puppynet],
    [2066098519157881736n, testnet_evm_ethereum_testnet_sepolia_xlayer_1],
    [16487132492576884721n, testnet_evm_cronos_zkevm_testnet_sepolia],
    [6802309497652714138n, testnet_evm_ethereum_testnet_goerli_zksync_1],
    [3842103497652714138n, testnet_evm_cronos_testnet_zkevm_1],
    [222782988166878823n, testnet_evm_hedera_testnet],
    [6898391096552792247n, testnet_evm_ethereum_testnet_sepolia_zksync_1],
    [2995292832068775165n, testnet_evm_cronos_testnet],
    [5061593697262339000n, testnet_evm_near_testnet],
    [2664363617261496610n, testnet_evm_ethereum_testnet_goerli_optimism_1],
    [7317911323415911000n, testnet_evm_areon_testnet],
    [5059197667603797935n, testnet_evm_janction_testnet_sepolia],
    [6260932437388305511n, testnet_evm_private_testnet_obsidian],
    [829525985033418733n, testnet_evm_ethereum_testnet_sepolia_mode_1],
    [2177900824115119161n, testnet_evm_bittensor_testnet],
    [4286062357653186312n, testnet_evm_hyperliquid_testnet],
    [2624132734533621656n, testnet_evm_kaia_testnet_kairos],
    [4459371029167934217n, testnet_evm_bittorrent_chain_testnet],
    [9284632837123596123n, testnet_evm_wemix_testnet],
    [4264732132125536123n, testnet_evm_core_testnet],
    [1948510578179542068n, testnet_evm_bitcoin_testnet_bsquared_1],
    [5361632739113536121n, testnet_evm_polkadot_testnet_moonbeam_moonbase],
    [14135854469784514356n, testnet_evm_ethereum_testnet_sepolia_unichain_1],
    [1216300075444106652n, testnet_evm_sei_testnet_atlantic],
    [3379446385462418246n, testnet_evm_geth_testnet],
    [11059667695644972511n, testnet_evm_ethereum_testnet_goerli_polygon_zkevm_1],
    [4237030917318060427n, testnet_evm_story_testnet],
    [10749384167430721561n, testnet_evm_mint_testnet],
    [6286293440461807648n, testnet_evm_metal_testnet],
    [4888058894222120000n, testnet_evm_bitcichain_testnet],
    [686603546605904534n, testnet_evm_ethereum_testnet_sepolia_soneium_1],
    [13116810400804392105n, testnet_evm_ronin_testnet_saigon],
    [3260900564719373474n, testnet_evm_private_testnet_granite],
    [6915682381028791124n, testnet_evm_private_testnet_andesite],
    [15513093881969820114n, testnet_evm_dtcc_testnet_andesite],
    [2333097300889804761n, testnet_evm_polkadot_testnet_centrifuge_altair],
    [12168171414969487009n, testnet_evm_memento_testnet],
    [2110537777356199208n, testnet_evm_kava_testnet],
    [5990477251245693094n, testnet_evm_ethereum_testnet_sepolia_kroma_1],
    [9488606126177218005n, testnet_evm_tac_testnet],
    [1654667687261492630n, testnet_evm_ethereum_testnet_sepolia_polygon_zkevm_1],
    [8901520481741771655n, testnet_evm_ethereum_testnet_holesky_fraxtal_1],
    [8304510386741731151n, testnet_evm_ethereum_testnet_holesky_morph_1],
    [1467223411771711614n, testnet_evm_bitcoin_testnet_botanix],
    [4905564228793744293n, testnet_evm_fantom_testnet],
    [5298399861320400553n, testnet_evm_ethereum_testnet_sepolia_lisk_1],
    [5299555114858065850n, testnet_evm_ethereum_testnet_sepolia_worldchain_1],
    [4168263376276232250n, testnet_evm_ethereum_testnet_goerli_mantle_1],
    [8236463271206331221n, testnet_evm_ethereum_testnet_sepolia_mantle_1],
    [13274425992935471758n, testnet_evm_binance_smart_chain_testnet_opbnb_1],
    [8911150974185440581n, testnet_evm_nexon_dev],
    [2443239559770384419n, testnet_evm_megaeth_testnet],
    [305104239123120457n, testnet_evm_nibiru_testnet],
    [344208382356656551n, testnet_evm_ondo_testnet],
    [1113014352258747600n, testnet_evm_neonlink_testnet],
    [3967220077692964309n, testnet_evm_plasma_testnet],
    [2183018362218727504n, testnet_evm_monad_testnet],
    [8871595565390010547n, testnet_evm_gnosis_chain_testnet_chiado],
    [16235373811196386733n, testnet_evm_abstract_testnet],
    [3486622437121596122n, testnet_evm_ethereum_testnet_sepolia_arbitrum_1_l3x_1],
    [4526165231216331901n, testnet_evm_ethereum_testnet_sepolia_immutable_zkevm_1],
    [16088006396410204581n, testnet_evm_0g_testnet_newton],
    [2131427466778448014n, testnet_evm_0g_testnet_galileo],
    [7717148896336251131n, testnet_evm_ethereum_testnet_holesky],
    [7759470850252068959n, testnet_evm_anvil_devnet],
    [9900119385908781505n, testnet_evm_apechain_testnet_curtis],
    [6827576821754315911n, testnet_evm_ethereum_testnet_sepolia_lens_1],
    [14767482510784806043n, testnet_evm_avalanche_testnet_fuji],
    [3552045678561919002n, testnet_evm_celo_testnet_alfajores],
    [8446413392851542429n, testnet_evm_private_testnet_opala],
    [13781831279385219069n, testnet_evm_zircuit_testnet_garfield],
    [4562743618362911021n, testnet_evm_ethereum_testnet_sepolia_zircuit_1],
    [13694007683517087973n, testnet_evm_superseed_testnet],
    [3676871237479449268n, testnet_evm_sonic_testnet_blaze],
    [1355246678561316402n, testnet_evm_ethereum_testnet_goerli_linea_1],
    [5719461335882077547n, testnet_evm_ethereum_testnet_sepolia_linea_1],
    [3777822886988675105n, testnet_evm_ethereum_testnet_sepolia_metis_1],
    [12532609583862916517n, testnet_evm_polygon_testnet_mumbai],
    [16281711391670634445n, testnet_evm_polygon_testnet_amoy],
    [7728255861635209484n, testnet_evm_berachain_testnet_bepolia],
    [8999465244383784164n, testnet_evm_berachain_testnet_bartio],
    [12336603543561911511n, testnet_evm_berachain_testnet_artio],
    [2285225387454015855n, testnet_evm_zero_g_testnet_galileo],
    [5790810961207155433n, testnet_evm_ethereum_testnet_goerli_base_1],
    [10344971235874465080n, testnet_evm_ethereum_testnet_sepolia_base_1],
    [3743020999916460931n, testnet_evm_plume_devnet],
    [13874588925447303949n, testnet_evm_plume_testnet_sepolia],
    [1910019406958449359n, testnet_evm_etherlink_testnet],
    [9090863410735740267n, testnet_evm_polygon_testnet_tatara],
    [7248756420937879088n, testnet_evm_ethereum_testnet_holesky_taiko_1],
    [7189150270347329685n, testnet_evm_mind_testnet],
    [3789623672476206327n, testnet_evm_bitcoin_testnet_bitlayer_1],
    [6101244977088475029n, testnet_evm_ethereum_testnet_goerli_arbitrum_1],
    [3478487238524512106n, testnet_evm_ethereum_testnet_sepolia_arbitrum_1],
    [4489326297382772450n, testnet_evm_private_testnet_mica],
    [1458281248224512906n, testnet_evm_avalanche_subnet_dexalot_testnet],
    [2279865765895943307n, testnet_evm_ethereum_testnet_sepolia_scroll_1],
    [7837562506228496256n, testnet_evm_avalanche_testnet_nexon],
    [5269261765892944301n, testnet_evm_bitcoin_testnet_merlin],
    [4012524741200567430n, testnet_evm_pharos_testnet],
    [4418231248214522936n, testnet_evm_ethereum_testnet_sepolia_polygon_validium_1],
    [16126893759944359622n, testnet_evm_hemi_testnet_sepolia],
    [9763904284804119144n, testnet_evm_ink_testnet_sepolia],
    [5535534526963509396n, testnet_evm_bitcoin_testnet_sepolia_bob_1],
    [5837261596322416298n, testnet_evm_zklink_nova_testnet],
    [7225665875429174318n, testnet_evm_codex_testnet],
    [10443705513486043421n, testnet_evm_ethereum_testnet_sepolia_arbitrum_1_treasure_1],
    [3676916124122457866n, testnet_evm_treasure_testnet_topaz],
    [945045181441419236n, testnet_evm_jovay_testnet],
    [16015286601757825753n, testnet_evm_ethereum_testnet_sepolia],
    [5224473277236331295n, testnet_evm_ethereum_testnet_sepolia_optimism_1],
    [2217764097022649312n, testnet_evm_neox_testnet_t4],
    [1467427327723633929n, testnet_evm_ethereum_testnet_sepolia_corn_1],
    [7060342227814389000n, testnet_evm_filecoin_testnet],
    [14684575664602284776n, testnet_evm_plume_testnet],
    [2027362563942762617n, testnet_evm_ethereum_testnet_sepolia_blast_1],
    [16244020411108056671n, testnet_evm_zora_testnet],
    [13231703482326770598n, testnet_evm_tron_testnet_shasta_evm],
    [13231703482326770600n, testnet_evm_tron_devnet_evm],
    [2052925811360307749n, testnet_evm_tron_testnet_nile_evm],
    [6302590918974934319n, testnet_solana_solana_testnet],
    [16423721717087811551n, testnet_solana_solana_devnet],
    [743186221051783445n, testnet_aptos_aptos_testnet],
    [4457093679053095497n, testnet_aptos_aptos_localnet],
    [9762610643973837292n, testnet_sui_sui_testnet],
    [18395503381733958356n, testnet_sui_sui_localnet],
    [1399300952838017768n, testnet_ton_ton_testnet],
    [13879075125137744094n, testnet_ton_ton_localnet],
    [13231703482326770597n, testnet_tron_tron_testnet_shasta],
    [13231703482326770599n, testnet_tron_tron_devnet],
    [2052925811360307740n, testnet_tron_tron_testnet_nile],
]);
// Optimized Maps for fast lookups by chain selector name
export const mainnetByName = new Map([
    ['ethereum-mainnet', mainnet_evm_ethereum_mainnet],
    ['ethereum-mainnet-optimism-1', mainnet_evm_ethereum_mainnet_optimism_1],
    ['cronos-mainnet', mainnet_evm_cronos_mainnet],
    ['rootstock-mainnet', mainnet_evm_rootstock_mainnet],
    ['telos-evm-mainnet', mainnet_evm_telos_evm_mainnet],
    ['polkadot-mainnet-darwinia', mainnet_evm_polkadot_mainnet_darwinia],
    ['xdc-mainnet', mainnet_evm_xdc_mainnet],
    ['coinex_smart_chain-mainnet', mainnet_evm_coinex_smart_chain_mainnet],
    ['binance_smart_chain-mainnet', mainnet_evm_binance_smart_chain_mainnet],
    ['gnosis_chain-mainnet', mainnet_evm_gnosis_chain_mainnet],
    ['velas-mainnet', mainnet_evm_velas_mainnet],
    ['shibarium-mainnet', mainnet_evm_shibarium_mainnet],
    ['ethereum-mainnet-unichain-1', mainnet_evm_ethereum_mainnet_unichain_1],
    ['polygon-mainnet', mainnet_evm_polygon_mainnet],
    ['monad-mainnet', mainnet_evm_monad_mainnet],
    ['sonic-mainnet', mainnet_evm_sonic_mainnet],
    ['ethereum-mainnet-hashkey-1', mainnet_evm_ethereum_mainnet_hashkey_1],
    ['mint-mainnet', mainnet_evm_mint_mainnet],
    ['ethereum-mainnet-xlayer-1', mainnet_evm_ethereum_mainnet_xlayer_1],
    ['bittorrent_chain-mainnet', mainnet_evm_bittorrent_chain_mainnet],
    ['binance_smart_chain-mainnet-opbnb-1', mainnet_evm_binance_smart_chain_mainnet_opbnb_1],
    ['bitcoin-mainnet-bsquared-1', mainnet_evm_bitcoin_mainnet_bsquared_1],
    ['mind-mainnet', mainnet_evm_mind_mainnet],
    ['lens-mainnet', mainnet_evm_lens_mainnet],
    ['tac-mainnet', mainnet_evm_tac_mainnet],
    ['fantom-mainnet', mainnet_evm_fantom_mainnet],
    ['fraxtal-mainnet', mainnet_evm_fraxtal_mainnet],
    ['ethereum-mainnet-kroma-1', mainnet_evm_ethereum_mainnet_kroma_1],
    ['neonlink-mainnet', mainnet_evm_neonlink_mainnet],
    ['hedera-mainnet', mainnet_evm_hedera_mainnet],
    ['filecoin-mainnet', mainnet_evm_filecoin_mainnet],
    ['ethereum-mainnet-zksync-1', mainnet_evm_ethereum_mainnet_zksync_1],
    ['cronos-zkevm-mainnet', mainnet_evm_cronos_zkevm_mainnet],
    ['near-mainnet', mainnet_evm_near_mainnet],
    ['areon-mainnet', mainnet_evm_areon_mainnet],
    ['ethereum-mainnet-worldchain-1', mainnet_evm_ethereum_mainnet_worldchain_1],
    ['polkadot-mainnet-astar', mainnet_evm_polkadot_mainnet_astar],
    ['janction-mainnet', mainnet_evm_janction_mainnet],
    ['bittensor-mainnet', mainnet_evm_bittensor_mainnet],
    ['hyperliquid-mainnet', mainnet_evm_hyperliquid_mainnet],
    ['conflux-mainnet', mainnet_evm_conflux_mainnet],
    ['ethereum-mainnet-metis-1', mainnet_evm_ethereum_mainnet_metis_1],
    ['ethereum-mainnet-polygon-zkevm-1', mainnet_evm_ethereum_mainnet_polygon_zkevm_1],
    ['wemix-mainnet', mainnet_evm_wemix_mainnet],
    ['core-mainnet', mainnet_evm_core_mainnet],
    ['lisk-mainnet', mainnet_evm_lisk_mainnet],
    ['polkadot-mainnet-moonbeam', mainnet_evm_polkadot_mainnet_moonbeam],
    ['kusama-mainnet-moonriver', mainnet_evm_kusama_mainnet_moonriver],
    ['sei-mainnet', mainnet_evm_sei_mainnet],
    ['metal-mainnet', mainnet_evm_metal_mainnet],
    ['soneium-mainnet', mainnet_evm_soneium_mainnet],
    ['bitcichain-mainnet', mainnet_evm_bitcichain_mainnet],
    ['ronin-mainnet', mainnet_evm_ronin_mainnet],
    ['polkadot-mainnet-centrifuge', mainnet_evm_polkadot_mainnet_centrifuge],
    ['kava-mainnet', mainnet_evm_kava_mainnet],
    ['abstract-mainnet', mainnet_evm_abstract_mainnet],
    ['morph-mainnet', mainnet_evm_morph_mainnet],
    ['bitcoin-mainnet-botanix', mainnet_evm_bitcoin_mainnet_botanix],
    ['ethereum-mainnet-astar-zkevm-1', mainnet_evm_ethereum_mainnet_astar_zkevm_1],
    ['bitcoin-merlin-mainnet', mainnet_evm_bitcoin_merlin_mainnet],
    ['ethereum-mainnet-mantle-1', mainnet_evm_ethereum_mainnet_mantle_1],
    ['superseed-mainnet', mainnet_evm_superseed_mainnet],
    ['nibiru-mainnet', mainnet_evm_nibiru_mainnet],
    ['zetachain-mainnet', mainnet_evm_zetachain_mainnet],
    ['kaia-mainnet', mainnet_evm_kaia_mainnet],
    ['ethereum-mainnet-base-1', mainnet_evm_ethereum_mainnet_base_1],
    ['plasma-mainnet', mainnet_evm_plasma_mainnet],
    ['ethereum-mainnet-arbitrum-1-l3x-1', mainnet_evm_ethereum_mainnet_arbitrum_1_l3x_1],
    ['ethereum-mainnet-immutable-zkevm-1', mainnet_evm_ethereum_mainnet_immutable_zkevm_1],
    ['0g-mainnet', mainnet_evm_0g_mainnet],
    ['apechain-mainnet', mainnet_evm_apechain_mainnet],
    ['ethereum-mainnet-mode-1', mainnet_evm_ethereum_mainnet_mode_1],
    ['ethereum-mainnet-arbitrum-1', mainnet_evm_ethereum_mainnet_arbitrum_1],
    ['celo-mainnet', mainnet_evm_celo_mainnet],
    ['etherlink-mainnet', mainnet_evm_etherlink_mainnet],
    ['hemi-mainnet', mainnet_evm_hemi_mainnet],
    ['avalanche-mainnet', mainnet_evm_avalanche_mainnet],
    ['neox-mainnet', mainnet_evm_neox_mainnet],
    ['ethereum-mainnet-zircuit-1', mainnet_evm_ethereum_mainnet_zircuit_1],
    ['memento-mainnet', mainnet_evm_memento_mainnet],
    ['ethereum-mainnet-ink-1', mainnet_evm_ethereum_mainnet_ink_1],
    ['ethereum-mainnet-linea-1', mainnet_evm_ethereum_mainnet_linea_1],
    ['nexon-mainnet-lith', mainnet_evm_nexon_mainnet_lith],
    ['bitcoin-mainnet-bob-1', mainnet_evm_bitcoin_mainnet_bob_1],
    ['treasure-mainnet', mainnet_evm_treasure_mainnet],
    ['nexon-mainnet-henesys', mainnet_evm_nexon_mainnet_henesys],
    ['berachain-mainnet', mainnet_evm_berachain_mainnet],
    ['codex-mainnet', mainnet_evm_codex_mainnet],
    ['ethereum-mainnet-blast-1', mainnet_evm_ethereum_mainnet_blast_1],
    ['plume-mainnet', mainnet_evm_plume_mainnet],
    ['ethereum-mainnet-taiko-1', mainnet_evm_ethereum_mainnet_taiko_1],
    ['bitcoin-mainnet-bitlayer-1', mainnet_evm_bitcoin_mainnet_bitlayer_1],
    ['avalanche-subnet-dexalot-mainnet', mainnet_evm_avalanche_subnet_dexalot_mainnet],
    ['ethereum-mainnet-scroll-1', mainnet_evm_ethereum_mainnet_scroll_1],
    ['polygon-mainnet-katana', mainnet_evm_polygon_mainnet_katana],
    ['nexon-qa', mainnet_evm_nexon_qa],
    ['zklink_nova-mainnet', mainnet_evm_zklink_nova_mainnet],
    ['nexon-stage', mainnet_evm_nexon_stage],
    ['ethereum-mainnet-arbitrum-1-treasure-1', mainnet_evm_ethereum_mainnet_arbitrum_1_treasure_1],
    ['zora-mainnet', mainnet_evm_zora_mainnet],
    ['corn-mainnet', mainnet_evm_corn_mainnet],
    ['tron-mainnet-evm', mainnet_evm_tron_mainnet_evm],
    ['solana-mainnet', mainnet_solana_solana_mainnet],
    ['aptos-mainnet', mainnet_aptos_aptos_mainnet],
    ['sui-mainnet', mainnet_sui_sui_mainnet],
    ['ton-mainnet', mainnet_ton_ton_mainnet],
    ['tron-mainnet', mainnet_tron_tron_mainnet],
]);
export const testnetByName = new Map([
    ['bitcoin-testnet-rootstock', testnet_evm_bitcoin_testnet_rootstock],
    ['telos-evm-testnet', testnet_evm_telos_evm_testnet],
    ['polkadot-testnet-darwinia-pangoro', testnet_evm_polkadot_testnet_darwinia_pangoro],
    ['xdc-testnet', testnet_evm_xdc_testnet],
    ['coinex_smart_chain-testnet', testnet_evm_coinex_smart_chain_testnet],
    ['polkadot-testnet-astar-shibuya', testnet_evm_polkadot_testnet_astar_shibuya],
    ['binance_smart_chain-testnet', testnet_evm_binance_smart_chain_testnet],
    ['velas-testnet', testnet_evm_velas_testnet],
    ['ethereum-testnet-sepolia-hashkey-1', testnet_evm_ethereum_testnet_sepolia_hashkey_1],
    ['shibarium-testnet-puppynet', testnet_evm_shibarium_testnet_puppynet],
    ['ethereum-testnet-sepolia-xlayer-1', testnet_evm_ethereum_testnet_sepolia_xlayer_1],
    ['cronos-zkevm-testnet-sepolia', testnet_evm_cronos_zkevm_testnet_sepolia],
    ['ethereum-testnet-goerli-zksync-1', testnet_evm_ethereum_testnet_goerli_zksync_1],
    ['cronos-testnet-zkevm-1', testnet_evm_cronos_testnet_zkevm_1],
    ['hedera-testnet', testnet_evm_hedera_testnet],
    ['ethereum-testnet-sepolia-zksync-1', testnet_evm_ethereum_testnet_sepolia_zksync_1],
    ['cronos-testnet', testnet_evm_cronos_testnet],
    ['near-testnet', testnet_evm_near_testnet],
    ['ethereum-testnet-goerli-optimism-1', testnet_evm_ethereum_testnet_goerli_optimism_1],
    ['areon-testnet', testnet_evm_areon_testnet],
    ['janction-testnet-sepolia', testnet_evm_janction_testnet_sepolia],
    ['private-testnet-obsidian', testnet_evm_private_testnet_obsidian],
    ['ethereum-testnet-sepolia-mode-1', testnet_evm_ethereum_testnet_sepolia_mode_1],
    ['bittensor-testnet', testnet_evm_bittensor_testnet],
    ['hyperliquid-testnet', testnet_evm_hyperliquid_testnet],
    ['kaia-testnet-kairos', testnet_evm_kaia_testnet_kairos],
    ['bittorrent_chain-testnet', testnet_evm_bittorrent_chain_testnet],
    ['wemix-testnet', testnet_evm_wemix_testnet],
    ['core-testnet', testnet_evm_core_testnet],
    ['bitcoin-testnet-bsquared-1', testnet_evm_bitcoin_testnet_bsquared_1],
    ['polkadot-testnet-moonbeam-moonbase', testnet_evm_polkadot_testnet_moonbeam_moonbase],
    ['ethereum-testnet-sepolia-unichain-1', testnet_evm_ethereum_testnet_sepolia_unichain_1],
    ['sei-testnet-atlantic', testnet_evm_sei_testnet_atlantic],
    ['geth-testnet', testnet_evm_geth_testnet],
    ['ethereum-testnet-goerli-polygon-zkevm-1', testnet_evm_ethereum_testnet_goerli_polygon_zkevm_1],
    ['story-testnet', testnet_evm_story_testnet],
    ['mint-testnet', testnet_evm_mint_testnet],
    ['metal-testnet', testnet_evm_metal_testnet],
    ['bitcichain-testnet', testnet_evm_bitcichain_testnet],
    ['ethereum-testnet-sepolia-soneium-1', testnet_evm_ethereum_testnet_sepolia_soneium_1],
    ['ronin-testnet-saigon', testnet_evm_ronin_testnet_saigon],
    ['private-testnet-granite', testnet_evm_private_testnet_granite],
    ['private-testnet-andesite', testnet_evm_private_testnet_andesite],
    ['dtcc-testnet-andesite', testnet_evm_dtcc_testnet_andesite],
    ['polkadot-testnet-centrifuge-altair', testnet_evm_polkadot_testnet_centrifuge_altair],
    ['memento-testnet', testnet_evm_memento_testnet],
    ['kava-testnet', testnet_evm_kava_testnet],
    ['ethereum-testnet-sepolia-kroma-1', testnet_evm_ethereum_testnet_sepolia_kroma_1],
    ['tac-testnet', testnet_evm_tac_testnet],
    [
        'ethereum-testnet-sepolia-polygon-zkevm-1',
        testnet_evm_ethereum_testnet_sepolia_polygon_zkevm_1,
    ],
    ['ethereum-testnet-holesky-fraxtal-1', testnet_evm_ethereum_testnet_holesky_fraxtal_1],
    ['ethereum-testnet-holesky-morph-1', testnet_evm_ethereum_testnet_holesky_morph_1],
    ['bitcoin-testnet-botanix', testnet_evm_bitcoin_testnet_botanix],
    ['fantom-testnet', testnet_evm_fantom_testnet],
    ['ethereum-testnet-sepolia-lisk-1', testnet_evm_ethereum_testnet_sepolia_lisk_1],
    ['ethereum-testnet-sepolia-worldchain-1', testnet_evm_ethereum_testnet_sepolia_worldchain_1],
    ['ethereum-testnet-goerli-mantle-1', testnet_evm_ethereum_testnet_goerli_mantle_1],
    ['ethereum-testnet-sepolia-mantle-1', testnet_evm_ethereum_testnet_sepolia_mantle_1],
    ['binance_smart_chain-testnet-opbnb-1', testnet_evm_binance_smart_chain_testnet_opbnb_1],
    ['nexon-dev', testnet_evm_nexon_dev],
    ['megaeth-testnet', testnet_evm_megaeth_testnet],
    ['nibiru-testnet', testnet_evm_nibiru_testnet],
    ['ondo-testnet', testnet_evm_ondo_testnet],
    ['neonlink-testnet', testnet_evm_neonlink_testnet],
    ['plasma-testnet', testnet_evm_plasma_testnet],
    ['monad-testnet', testnet_evm_monad_testnet],
    ['gnosis_chain-testnet-chiado', testnet_evm_gnosis_chain_testnet_chiado],
    ['abstract-testnet', testnet_evm_abstract_testnet],
    [
        'ethereum-testnet-sepolia-arbitrum-1-l3x-1',
        testnet_evm_ethereum_testnet_sepolia_arbitrum_1_l3x_1,
    ],
    [
        'ethereum-testnet-sepolia-immutable-zkevm-1',
        testnet_evm_ethereum_testnet_sepolia_immutable_zkevm_1,
    ],
    ['0g-testnet-newton', testnet_evm_0g_testnet_newton],
    ['0g-testnet-galileo', testnet_evm_0g_testnet_galileo],
    ['ethereum-testnet-holesky', testnet_evm_ethereum_testnet_holesky],
    ['anvil-devnet', testnet_evm_anvil_devnet],
    ['apechain-testnet-curtis', testnet_evm_apechain_testnet_curtis],
    ['ethereum-testnet-sepolia-lens-1', testnet_evm_ethereum_testnet_sepolia_lens_1],
    ['avalanche-testnet-fuji', testnet_evm_avalanche_testnet_fuji],
    ['celo-testnet-alfajores', testnet_evm_celo_testnet_alfajores],
    ['private-testnet-opala', testnet_evm_private_testnet_opala],
    ['zircuit-testnet-garfield', testnet_evm_zircuit_testnet_garfield],
    ['ethereum-testnet-sepolia-zircuit-1', testnet_evm_ethereum_testnet_sepolia_zircuit_1],
    ['superseed-testnet', testnet_evm_superseed_testnet],
    ['sonic-testnet-blaze', testnet_evm_sonic_testnet_blaze],
    ['ethereum-testnet-goerli-linea-1', testnet_evm_ethereum_testnet_goerli_linea_1],
    ['ethereum-testnet-sepolia-linea-1', testnet_evm_ethereum_testnet_sepolia_linea_1],
    ['ethereum-testnet-sepolia-metis-1', testnet_evm_ethereum_testnet_sepolia_metis_1],
    ['polygon-testnet-mumbai', testnet_evm_polygon_testnet_mumbai],
    ['polygon-testnet-amoy', testnet_evm_polygon_testnet_amoy],
    ['berachain-testnet-bepolia', testnet_evm_berachain_testnet_bepolia],
    ['berachain-testnet-bartio', testnet_evm_berachain_testnet_bartio],
    ['berachain-testnet-artio', testnet_evm_berachain_testnet_artio],
    ['zero-g-testnet-galileo', testnet_evm_zero_g_testnet_galileo],
    ['ethereum-testnet-goerli-base-1', testnet_evm_ethereum_testnet_goerli_base_1],
    ['ethereum-testnet-sepolia-base-1', testnet_evm_ethereum_testnet_sepolia_base_1],
    ['plume-devnet', testnet_evm_plume_devnet],
    ['plume-testnet-sepolia', testnet_evm_plume_testnet_sepolia],
    ['etherlink-testnet', testnet_evm_etherlink_testnet],
    ['polygon-testnet-tatara', testnet_evm_polygon_testnet_tatara],
    ['ethereum-testnet-holesky-taiko-1', testnet_evm_ethereum_testnet_holesky_taiko_1],
    ['mind-testnet', testnet_evm_mind_testnet],
    ['bitcoin-testnet-bitlayer-1', testnet_evm_bitcoin_testnet_bitlayer_1],
    ['ethereum-testnet-goerli-arbitrum-1', testnet_evm_ethereum_testnet_goerli_arbitrum_1],
    ['ethereum-testnet-sepolia-arbitrum-1', testnet_evm_ethereum_testnet_sepolia_arbitrum_1],
    ['private-testnet-mica', testnet_evm_private_testnet_mica],
    ['avalanche-subnet-dexalot-testnet', testnet_evm_avalanche_subnet_dexalot_testnet],
    ['ethereum-testnet-sepolia-scroll-1', testnet_evm_ethereum_testnet_sepolia_scroll_1],
    ['avalanche-testnet-nexon', testnet_evm_avalanche_testnet_nexon],
    ['bitcoin-testnet-merlin', testnet_evm_bitcoin_testnet_merlin],
    ['pharos-testnet', testnet_evm_pharos_testnet],
    [
        'ethereum-testnet-sepolia-polygon-validium-1',
        testnet_evm_ethereum_testnet_sepolia_polygon_validium_1,
    ],
    ['hemi-testnet-sepolia', testnet_evm_hemi_testnet_sepolia],
    ['ink-testnet-sepolia', testnet_evm_ink_testnet_sepolia],
    ['bitcoin-testnet-sepolia-bob-1', testnet_evm_bitcoin_testnet_sepolia_bob_1],
    ['zklink_nova-testnet', testnet_evm_zklink_nova_testnet],
    ['codex-testnet', testnet_evm_codex_testnet],
    [
        'ethereum-testnet-sepolia-arbitrum-1-treasure-1',
        testnet_evm_ethereum_testnet_sepolia_arbitrum_1_treasure_1,
    ],
    ['treasure-testnet-topaz', testnet_evm_treasure_testnet_topaz],
    ['jovay-testnet', testnet_evm_jovay_testnet],
    ['ethereum-testnet-sepolia', testnet_evm_ethereum_testnet_sepolia],
    ['ethereum-testnet-sepolia-optimism-1', testnet_evm_ethereum_testnet_sepolia_optimism_1],
    ['neox-testnet-t4', testnet_evm_neox_testnet_t4],
    ['ethereum-testnet-sepolia-corn-1', testnet_evm_ethereum_testnet_sepolia_corn_1],
    ['filecoin-testnet', testnet_evm_filecoin_testnet],
    ['plume-testnet', testnet_evm_plume_testnet],
    ['ethereum-testnet-sepolia-blast-1', testnet_evm_ethereum_testnet_sepolia_blast_1],
    ['zora-testnet', testnet_evm_zora_testnet],
    ['tron-testnet-shasta-evm', testnet_evm_tron_testnet_shasta_evm],
    ['tron-devnet-evm', testnet_evm_tron_devnet_evm],
    ['tron-testnet-nile-evm', testnet_evm_tron_testnet_nile_evm],
    ['solana-testnet', testnet_solana_solana_testnet],
    ['solana-devnet', testnet_solana_solana_devnet],
    ['aptos-testnet', testnet_aptos_aptos_testnet],
    ['aptos-localnet', testnet_aptos_aptos_localnet],
    ['sui-testnet', testnet_sui_sui_testnet],
    ['sui-localnet', testnet_sui_sui_localnet],
    ['ton-testnet', testnet_ton_ton_testnet],
    ['ton-localnet', testnet_ton_ton_localnet],
    ['tron-testnet-shasta', testnet_tron_tron_testnet_shasta],
    ['tron-devnet', testnet_tron_tron_devnet],
    ['tron-testnet-nile', testnet_tron_tron_testnet_nile],
]);
// Maps by family and network type for chain selector lookups
export const mainnetBySelectorByFamily = {
    evm: new Map([
        [5009297550715157269n, mainnet_evm_ethereum_mainnet],
        [3734403246176062136n, mainnet_evm_ethereum_mainnet_optimism_1],
        [1456215246176062136n, mainnet_evm_cronos_mainnet],
        [11964252391146578476n, mainnet_evm_rootstock_mainnet],
        [1477345371608778000n, mainnet_evm_telos_evm_mainnet],
        [8866418665544333000n, mainnet_evm_polkadot_mainnet_darwinia],
        [17673274061779414707n, mainnet_evm_xdc_mainnet],
        [1761333065194157300n, mainnet_evm_coinex_smart_chain_mainnet],
        [11344663589394136015n, mainnet_evm_binance_smart_chain_mainnet],
        [465200170687744372n, mainnet_evm_gnosis_chain_mainnet],
        [374210358663784372n, mainnet_evm_velas_mainnet],
        [3993510008929295315n, mainnet_evm_shibarium_mainnet],
        [1923510103922296319n, mainnet_evm_ethereum_mainnet_unichain_1],
        [4051577828743386545n, mainnet_evm_polygon_mainnet],
        [8481857512324358265n, mainnet_evm_monad_mainnet],
        [1673871237479749969n, mainnet_evm_sonic_mainnet],
        [7613811247471741961n, mainnet_evm_ethereum_mainnet_hashkey_1],
        [17164792800244661392n, mainnet_evm_mint_mainnet],
        [3016212468291539606n, mainnet_evm_ethereum_mainnet_xlayer_1],
        [3776006016387883143n, mainnet_evm_bittorrent_chain_mainnet],
        [465944652040885897n, mainnet_evm_binance_smart_chain_mainnet_opbnb_1],
        [5406759801798337480n, mainnet_evm_bitcoin_mainnet_bsquared_1],
        [11690709103138290329n, mainnet_evm_mind_mainnet],
        [5608378062013572713n, mainnet_evm_lens_mainnet],
        [5936861837188149645n, mainnet_evm_tac_mainnet],
        [3768048213127883732n, mainnet_evm_fantom_mainnet],
        [1462016016387883143n, mainnet_evm_fraxtal_mainnet],
        [3719320017875267166n, mainnet_evm_ethereum_mainnet_kroma_1],
        [8239338020728974000n, mainnet_evm_neonlink_mainnet],
        [3229138320728879060n, mainnet_evm_hedera_mainnet],
        [4561443241176882990n, mainnet_evm_filecoin_mainnet],
        [1562403441176082196n, mainnet_evm_ethereum_mainnet_zksync_1],
        [8788096068760390840n, mainnet_evm_cronos_zkevm_mainnet],
        [2039744413822257700n, mainnet_evm_near_mainnet],
        [1939936305787790600n, mainnet_evm_areon_mainnet],
        [2049429975587534727n, mainnet_evm_ethereum_mainnet_worldchain_1],
        [6422105447186081193n, mainnet_evm_polkadot_mainnet_astar],
        [9107126442626377432n, mainnet_evm_janction_mainnet],
        [2135107236357186872n, mainnet_evm_bittensor_mainnet],
        [2442541497099098535n, mainnet_evm_hyperliquid_mainnet],
        [3358365939762719202n, mainnet_evm_conflux_mainnet],
        [8805746078405598895n, mainnet_evm_ethereum_mainnet_metis_1],
        [4348158687435793198n, mainnet_evm_ethereum_mainnet_polygon_zkevm_1],
        [5142893604156789321n, mainnet_evm_wemix_mainnet],
        [1224752112135636129n, mainnet_evm_core_mainnet],
        [15293031020466096408n, mainnet_evm_lisk_mainnet],
        [1252863800116739621n, mainnet_evm_polkadot_mainnet_moonbeam],
        [1355020143337428062n, mainnet_evm_kusama_mainnet_moonriver],
        [9027416829622342829n, mainnet_evm_sei_mainnet],
        [13447077090413146373n, mainnet_evm_metal_mainnet],
        [12505351618335765396n, mainnet_evm_soneium_mainnet],
        [4874388048629246000n, mainnet_evm_bitcichain_mainnet],
        [6916147374840168594n, mainnet_evm_ronin_mainnet],
        [8175830712062617656n, mainnet_evm_polkadot_mainnet_centrifuge],
        [7550000543357438061n, mainnet_evm_kava_mainnet],
        [3577778157919314504n, mainnet_evm_abstract_mainnet],
        [18164309074156128038n, mainnet_evm_morph_mainnet],
        [4560701533377838164n, mainnet_evm_bitcoin_mainnet_botanix],
        [1540201334317828111n, mainnet_evm_ethereum_mainnet_astar_zkevm_1],
        [241851231317828981n, mainnet_evm_bitcoin_merlin_mainnet],
        [1556008542357238666n, mainnet_evm_ethereum_mainnet_mantle_1],
        [470401360549526817n, mainnet_evm_superseed_mainnet],
        [17349189558768828726n, mainnet_evm_nibiru_mainnet],
        [10817664450262215148n, mainnet_evm_zetachain_mainnet],
        [9813823125703490621n, mainnet_evm_kaia_mainnet],
        [15971525489660198786n, mainnet_evm_ethereum_mainnet_base_1],
        [9335212494177455608n, mainnet_evm_plasma_mainnet],
        [3162193654116181371n, mainnet_evm_ethereum_mainnet_arbitrum_1_l3x_1],
        [1237925231416731909n, mainnet_evm_ethereum_mainnet_immutable_zkevm_1],
        [4426351306075016396n, mainnet_evm_0g_mainnet],
        [14894068710063348487n, mainnet_evm_apechain_mainnet],
        [7264351850409363825n, mainnet_evm_ethereum_mainnet_mode_1],
        [4949039107694359620n, mainnet_evm_ethereum_mainnet_arbitrum_1],
        [1346049177634351622n, mainnet_evm_celo_mainnet],
        [13624601974233774587n, mainnet_evm_etherlink_mainnet],
        [1804312132722180201n, mainnet_evm_hemi_mainnet],
        [6433500567565415381n, mainnet_evm_avalanche_mainnet],
        [7222032299962346917n, mainnet_evm_neox_mainnet],
        [17198166215261833993n, mainnet_evm_ethereum_mainnet_zircuit_1],
        [6473245816409426016n, mainnet_evm_memento_mainnet],
        [3461204551265785888n, mainnet_evm_ethereum_mainnet_ink_1],
        [4627098889531055414n, mainnet_evm_ethereum_mainnet_linea_1],
        [15758750456714168963n, mainnet_evm_nexon_mainnet_lith],
        [3849287863852499584n, mainnet_evm_bitcoin_mainnet_bob_1],
        [5214452172935136222n, mainnet_evm_treasure_mainnet],
        [12657445206920369324n, mainnet_evm_nexon_mainnet_henesys],
        [1294465214383781161n, mainnet_evm_berachain_mainnet],
        [9478124434908827753n, mainnet_evm_codex_mainnet],
        [4411394078118774322n, mainnet_evm_ethereum_mainnet_blast_1],
        [17912061998839310979n, mainnet_evm_plume_mainnet],
        [16468599424800719238n, mainnet_evm_ethereum_mainnet_taiko_1],
        [7937294810946806131n, mainnet_evm_bitcoin_mainnet_bitlayer_1],
        [5463201557265485081n, mainnet_evm_avalanche_subnet_dexalot_mainnet],
        [13204309965629103672n, mainnet_evm_ethereum_mainnet_scroll_1],
        [2459028469735686113n, mainnet_evm_polygon_mainnet_katana],
        [14632960069656270105n, mainnet_evm_nexon_qa],
        [4350319965322101699n, mainnet_evm_zklink_nova_mainnet],
        [5556806327594153475n, mainnet_evm_nexon_stage],
        [1010349088906777999n, mainnet_evm_ethereum_mainnet_arbitrum_1_treasure_1],
        [3555797439612589184n, mainnet_evm_zora_mainnet],
        [9043146809313071210n, mainnet_evm_corn_mainnet],
        [1546563616611573946n, mainnet_evm_tron_mainnet_evm],
    ]),
    solana: new Map([[124615329519749607n, mainnet_solana_solana_mainnet]]),
    aptos: new Map([[4741433654826277614n, mainnet_aptos_aptos_mainnet]]),
    sui: new Map([[17529533435026248318n, mainnet_sui_sui_mainnet]]),
    ton: new Map([[16448340667252469081n, mainnet_ton_ton_mainnet]]),
    tron: new Map([[1546563616611573945n, mainnet_tron_tron_mainnet]]),
};
export const testnetBySelectorByFamily = {
    evm: new Map([
        [8953668971247136127n, testnet_evm_bitcoin_testnet_rootstock],
        [729797994450396300n, testnet_evm_telos_evm_testnet],
        [4340886533089894000n, testnet_evm_polkadot_testnet_darwinia_pangoro],
        [3017758115101368649n, testnet_evm_xdc_testnet],
        [8955032871639343000n, testnet_evm_coinex_smart_chain_testnet],
        [6955638871347136141n, testnet_evm_polkadot_testnet_astar_shibuya],
        [13264668187771770619n, testnet_evm_binance_smart_chain_testnet],
        [572210378683744374n, testnet_evm_velas_testnet],
        [4356164186791070119n, testnet_evm_ethereum_testnet_sepolia_hashkey_1],
        [17833296867764334567n, testnet_evm_shibarium_testnet_puppynet],
        [2066098519157881736n, testnet_evm_ethereum_testnet_sepolia_xlayer_1],
        [16487132492576884721n, testnet_evm_cronos_zkevm_testnet_sepolia],
        [6802309497652714138n, testnet_evm_ethereum_testnet_goerli_zksync_1],
        [3842103497652714138n, testnet_evm_cronos_testnet_zkevm_1],
        [222782988166878823n, testnet_evm_hedera_testnet],
        [6898391096552792247n, testnet_evm_ethereum_testnet_sepolia_zksync_1],
        [2995292832068775165n, testnet_evm_cronos_testnet],
        [5061593697262339000n, testnet_evm_near_testnet],
        [2664363617261496610n, testnet_evm_ethereum_testnet_goerli_optimism_1],
        [7317911323415911000n, testnet_evm_areon_testnet],
        [5059197667603797935n, testnet_evm_janction_testnet_sepolia],
        [6260932437388305511n, testnet_evm_private_testnet_obsidian],
        [829525985033418733n, testnet_evm_ethereum_testnet_sepolia_mode_1],
        [2177900824115119161n, testnet_evm_bittensor_testnet],
        [4286062357653186312n, testnet_evm_hyperliquid_testnet],
        [2624132734533621656n, testnet_evm_kaia_testnet_kairos],
        [4459371029167934217n, testnet_evm_bittorrent_chain_testnet],
        [9284632837123596123n, testnet_evm_wemix_testnet],
        [4264732132125536123n, testnet_evm_core_testnet],
        [1948510578179542068n, testnet_evm_bitcoin_testnet_bsquared_1],
        [5361632739113536121n, testnet_evm_polkadot_testnet_moonbeam_moonbase],
        [14135854469784514356n, testnet_evm_ethereum_testnet_sepolia_unichain_1],
        [1216300075444106652n, testnet_evm_sei_testnet_atlantic],
        [3379446385462418246n, testnet_evm_geth_testnet],
        [11059667695644972511n, testnet_evm_ethereum_testnet_goerli_polygon_zkevm_1],
        [4237030917318060427n, testnet_evm_story_testnet],
        [10749384167430721561n, testnet_evm_mint_testnet],
        [6286293440461807648n, testnet_evm_metal_testnet],
        [4888058894222120000n, testnet_evm_bitcichain_testnet],
        [686603546605904534n, testnet_evm_ethereum_testnet_sepolia_soneium_1],
        [13116810400804392105n, testnet_evm_ronin_testnet_saigon],
        [3260900564719373474n, testnet_evm_private_testnet_granite],
        [6915682381028791124n, testnet_evm_private_testnet_andesite],
        [15513093881969820114n, testnet_evm_dtcc_testnet_andesite],
        [2333097300889804761n, testnet_evm_polkadot_testnet_centrifuge_altair],
        [12168171414969487009n, testnet_evm_memento_testnet],
        [2110537777356199208n, testnet_evm_kava_testnet],
        [5990477251245693094n, testnet_evm_ethereum_testnet_sepolia_kroma_1],
        [9488606126177218005n, testnet_evm_tac_testnet],
        [1654667687261492630n, testnet_evm_ethereum_testnet_sepolia_polygon_zkevm_1],
        [8901520481741771655n, testnet_evm_ethereum_testnet_holesky_fraxtal_1],
        [8304510386741731151n, testnet_evm_ethereum_testnet_holesky_morph_1],
        [1467223411771711614n, testnet_evm_bitcoin_testnet_botanix],
        [4905564228793744293n, testnet_evm_fantom_testnet],
        [5298399861320400553n, testnet_evm_ethereum_testnet_sepolia_lisk_1],
        [5299555114858065850n, testnet_evm_ethereum_testnet_sepolia_worldchain_1],
        [4168263376276232250n, testnet_evm_ethereum_testnet_goerli_mantle_1],
        [8236463271206331221n, testnet_evm_ethereum_testnet_sepolia_mantle_1],
        [13274425992935471758n, testnet_evm_binance_smart_chain_testnet_opbnb_1],
        [8911150974185440581n, testnet_evm_nexon_dev],
        [2443239559770384419n, testnet_evm_megaeth_testnet],
        [305104239123120457n, testnet_evm_nibiru_testnet],
        [344208382356656551n, testnet_evm_ondo_testnet],
        [1113014352258747600n, testnet_evm_neonlink_testnet],
        [3967220077692964309n, testnet_evm_plasma_testnet],
        [2183018362218727504n, testnet_evm_monad_testnet],
        [8871595565390010547n, testnet_evm_gnosis_chain_testnet_chiado],
        [16235373811196386733n, testnet_evm_abstract_testnet],
        [3486622437121596122n, testnet_evm_ethereum_testnet_sepolia_arbitrum_1_l3x_1],
        [4526165231216331901n, testnet_evm_ethereum_testnet_sepolia_immutable_zkevm_1],
        [16088006396410204581n, testnet_evm_0g_testnet_newton],
        [2131427466778448014n, testnet_evm_0g_testnet_galileo],
        [7717148896336251131n, testnet_evm_ethereum_testnet_holesky],
        [7759470850252068959n, testnet_evm_anvil_devnet],
        [9900119385908781505n, testnet_evm_apechain_testnet_curtis],
        [6827576821754315911n, testnet_evm_ethereum_testnet_sepolia_lens_1],
        [14767482510784806043n, testnet_evm_avalanche_testnet_fuji],
        [3552045678561919002n, testnet_evm_celo_testnet_alfajores],
        [8446413392851542429n, testnet_evm_private_testnet_opala],
        [13781831279385219069n, testnet_evm_zircuit_testnet_garfield],
        [4562743618362911021n, testnet_evm_ethereum_testnet_sepolia_zircuit_1],
        [13694007683517087973n, testnet_evm_superseed_testnet],
        [3676871237479449268n, testnet_evm_sonic_testnet_blaze],
        [1355246678561316402n, testnet_evm_ethereum_testnet_goerli_linea_1],
        [5719461335882077547n, testnet_evm_ethereum_testnet_sepolia_linea_1],
        [3777822886988675105n, testnet_evm_ethereum_testnet_sepolia_metis_1],
        [12532609583862916517n, testnet_evm_polygon_testnet_mumbai],
        [16281711391670634445n, testnet_evm_polygon_testnet_amoy],
        [7728255861635209484n, testnet_evm_berachain_testnet_bepolia],
        [8999465244383784164n, testnet_evm_berachain_testnet_bartio],
        [12336603543561911511n, testnet_evm_berachain_testnet_artio],
        [2285225387454015855n, testnet_evm_zero_g_testnet_galileo],
        [5790810961207155433n, testnet_evm_ethereum_testnet_goerli_base_1],
        [10344971235874465080n, testnet_evm_ethereum_testnet_sepolia_base_1],
        [3743020999916460931n, testnet_evm_plume_devnet],
        [13874588925447303949n, testnet_evm_plume_testnet_sepolia],
        [1910019406958449359n, testnet_evm_etherlink_testnet],
        [9090863410735740267n, testnet_evm_polygon_testnet_tatara],
        [7248756420937879088n, testnet_evm_ethereum_testnet_holesky_taiko_1],
        [7189150270347329685n, testnet_evm_mind_testnet],
        [3789623672476206327n, testnet_evm_bitcoin_testnet_bitlayer_1],
        [6101244977088475029n, testnet_evm_ethereum_testnet_goerli_arbitrum_1],
        [3478487238524512106n, testnet_evm_ethereum_testnet_sepolia_arbitrum_1],
        [4489326297382772450n, testnet_evm_private_testnet_mica],
        [1458281248224512906n, testnet_evm_avalanche_subnet_dexalot_testnet],
        [2279865765895943307n, testnet_evm_ethereum_testnet_sepolia_scroll_1],
        [7837562506228496256n, testnet_evm_avalanche_testnet_nexon],
        [5269261765892944301n, testnet_evm_bitcoin_testnet_merlin],
        [4012524741200567430n, testnet_evm_pharos_testnet],
        [4418231248214522936n, testnet_evm_ethereum_testnet_sepolia_polygon_validium_1],
        [16126893759944359622n, testnet_evm_hemi_testnet_sepolia],
        [9763904284804119144n, testnet_evm_ink_testnet_sepolia],
        [5535534526963509396n, testnet_evm_bitcoin_testnet_sepolia_bob_1],
        [5837261596322416298n, testnet_evm_zklink_nova_testnet],
        [7225665875429174318n, testnet_evm_codex_testnet],
        [10443705513486043421n, testnet_evm_ethereum_testnet_sepolia_arbitrum_1_treasure_1],
        [3676916124122457866n, testnet_evm_treasure_testnet_topaz],
        [945045181441419236n, testnet_evm_jovay_testnet],
        [16015286601757825753n, testnet_evm_ethereum_testnet_sepolia],
        [5224473277236331295n, testnet_evm_ethereum_testnet_sepolia_optimism_1],
        [2217764097022649312n, testnet_evm_neox_testnet_t4],
        [1467427327723633929n, testnet_evm_ethereum_testnet_sepolia_corn_1],
        [7060342227814389000n, testnet_evm_filecoin_testnet],
        [14684575664602284776n, testnet_evm_plume_testnet],
        [2027362563942762617n, testnet_evm_ethereum_testnet_sepolia_blast_1],
        [16244020411108056671n, testnet_evm_zora_testnet],
        [13231703482326770598n, testnet_evm_tron_testnet_shasta_evm],
        [13231703482326770600n, testnet_evm_tron_devnet_evm],
        [2052925811360307749n, testnet_evm_tron_testnet_nile_evm],
    ]),
    solana: new Map([
        [6302590918974934319n, testnet_solana_solana_testnet],
        [16423721717087811551n, testnet_solana_solana_devnet],
    ]),
    aptos: new Map([
        [743186221051783445n, testnet_aptos_aptos_testnet],
        [4457093679053095497n, testnet_aptos_aptos_localnet],
    ]),
    sui: new Map([
        [9762610643973837292n, testnet_sui_sui_testnet],
        [18395503381733958356n, testnet_sui_sui_localnet],
    ]),
    ton: new Map([
        [1399300952838017768n, testnet_ton_ton_testnet],
        [13879075125137744094n, testnet_ton_ton_localnet],
    ]),
    tron: new Map([
        [13231703482326770597n, testnet_tron_tron_testnet_shasta],
        [13231703482326770599n, testnet_tron_tron_devnet],
        [2052925811360307740n, testnet_tron_tron_testnet_nile],
    ]),
};
// Maps by family and network type for name lookups
export const mainnetByNameByFamily = {
    evm: new Map([
        ['ethereum-mainnet', mainnet_evm_ethereum_mainnet],
        ['ethereum-mainnet-optimism-1', mainnet_evm_ethereum_mainnet_optimism_1],
        ['cronos-mainnet', mainnet_evm_cronos_mainnet],
        ['rootstock-mainnet', mainnet_evm_rootstock_mainnet],
        ['telos-evm-mainnet', mainnet_evm_telos_evm_mainnet],
        ['polkadot-mainnet-darwinia', mainnet_evm_polkadot_mainnet_darwinia],
        ['xdc-mainnet', mainnet_evm_xdc_mainnet],
        ['coinex_smart_chain-mainnet', mainnet_evm_coinex_smart_chain_mainnet],
        ['binance_smart_chain-mainnet', mainnet_evm_binance_smart_chain_mainnet],
        ['gnosis_chain-mainnet', mainnet_evm_gnosis_chain_mainnet],
        ['velas-mainnet', mainnet_evm_velas_mainnet],
        ['shibarium-mainnet', mainnet_evm_shibarium_mainnet],
        ['ethereum-mainnet-unichain-1', mainnet_evm_ethereum_mainnet_unichain_1],
        ['polygon-mainnet', mainnet_evm_polygon_mainnet],
        ['monad-mainnet', mainnet_evm_monad_mainnet],
        ['sonic-mainnet', mainnet_evm_sonic_mainnet],
        ['ethereum-mainnet-hashkey-1', mainnet_evm_ethereum_mainnet_hashkey_1],
        ['mint-mainnet', mainnet_evm_mint_mainnet],
        ['ethereum-mainnet-xlayer-1', mainnet_evm_ethereum_mainnet_xlayer_1],
        ['bittorrent_chain-mainnet', mainnet_evm_bittorrent_chain_mainnet],
        ['binance_smart_chain-mainnet-opbnb-1', mainnet_evm_binance_smart_chain_mainnet_opbnb_1],
        ['bitcoin-mainnet-bsquared-1', mainnet_evm_bitcoin_mainnet_bsquared_1],
        ['mind-mainnet', mainnet_evm_mind_mainnet],
        ['lens-mainnet', mainnet_evm_lens_mainnet],
        ['tac-mainnet', mainnet_evm_tac_mainnet],
        ['fantom-mainnet', mainnet_evm_fantom_mainnet],
        ['fraxtal-mainnet', mainnet_evm_fraxtal_mainnet],
        ['ethereum-mainnet-kroma-1', mainnet_evm_ethereum_mainnet_kroma_1],
        ['neonlink-mainnet', mainnet_evm_neonlink_mainnet],
        ['hedera-mainnet', mainnet_evm_hedera_mainnet],
        ['filecoin-mainnet', mainnet_evm_filecoin_mainnet],
        ['ethereum-mainnet-zksync-1', mainnet_evm_ethereum_mainnet_zksync_1],
        ['cronos-zkevm-mainnet', mainnet_evm_cronos_zkevm_mainnet],
        ['near-mainnet', mainnet_evm_near_mainnet],
        ['areon-mainnet', mainnet_evm_areon_mainnet],
        ['ethereum-mainnet-worldchain-1', mainnet_evm_ethereum_mainnet_worldchain_1],
        ['polkadot-mainnet-astar', mainnet_evm_polkadot_mainnet_astar],
        ['janction-mainnet', mainnet_evm_janction_mainnet],
        ['bittensor-mainnet', mainnet_evm_bittensor_mainnet],
        ['hyperliquid-mainnet', mainnet_evm_hyperliquid_mainnet],
        ['conflux-mainnet', mainnet_evm_conflux_mainnet],
        ['ethereum-mainnet-metis-1', mainnet_evm_ethereum_mainnet_metis_1],
        ['ethereum-mainnet-polygon-zkevm-1', mainnet_evm_ethereum_mainnet_polygon_zkevm_1],
        ['wemix-mainnet', mainnet_evm_wemix_mainnet],
        ['core-mainnet', mainnet_evm_core_mainnet],
        ['lisk-mainnet', mainnet_evm_lisk_mainnet],
        ['polkadot-mainnet-moonbeam', mainnet_evm_polkadot_mainnet_moonbeam],
        ['kusama-mainnet-moonriver', mainnet_evm_kusama_mainnet_moonriver],
        ['sei-mainnet', mainnet_evm_sei_mainnet],
        ['metal-mainnet', mainnet_evm_metal_mainnet],
        ['soneium-mainnet', mainnet_evm_soneium_mainnet],
        ['bitcichain-mainnet', mainnet_evm_bitcichain_mainnet],
        ['ronin-mainnet', mainnet_evm_ronin_mainnet],
        ['polkadot-mainnet-centrifuge', mainnet_evm_polkadot_mainnet_centrifuge],
        ['kava-mainnet', mainnet_evm_kava_mainnet],
        ['abstract-mainnet', mainnet_evm_abstract_mainnet],
        ['morph-mainnet', mainnet_evm_morph_mainnet],
        ['bitcoin-mainnet-botanix', mainnet_evm_bitcoin_mainnet_botanix],
        ['ethereum-mainnet-astar-zkevm-1', mainnet_evm_ethereum_mainnet_astar_zkevm_1],
        ['bitcoin-merlin-mainnet', mainnet_evm_bitcoin_merlin_mainnet],
        ['ethereum-mainnet-mantle-1', mainnet_evm_ethereum_mainnet_mantle_1],
        ['superseed-mainnet', mainnet_evm_superseed_mainnet],
        ['nibiru-mainnet', mainnet_evm_nibiru_mainnet],
        ['zetachain-mainnet', mainnet_evm_zetachain_mainnet],
        ['kaia-mainnet', mainnet_evm_kaia_mainnet],
        ['ethereum-mainnet-base-1', mainnet_evm_ethereum_mainnet_base_1],
        ['plasma-mainnet', mainnet_evm_plasma_mainnet],
        ['ethereum-mainnet-arbitrum-1-l3x-1', mainnet_evm_ethereum_mainnet_arbitrum_1_l3x_1],
        ['ethereum-mainnet-immutable-zkevm-1', mainnet_evm_ethereum_mainnet_immutable_zkevm_1],
        ['0g-mainnet', mainnet_evm_0g_mainnet],
        ['apechain-mainnet', mainnet_evm_apechain_mainnet],
        ['ethereum-mainnet-mode-1', mainnet_evm_ethereum_mainnet_mode_1],
        ['ethereum-mainnet-arbitrum-1', mainnet_evm_ethereum_mainnet_arbitrum_1],
        ['celo-mainnet', mainnet_evm_celo_mainnet],
        ['etherlink-mainnet', mainnet_evm_etherlink_mainnet],
        ['hemi-mainnet', mainnet_evm_hemi_mainnet],
        ['avalanche-mainnet', mainnet_evm_avalanche_mainnet],
        ['neox-mainnet', mainnet_evm_neox_mainnet],
        ['ethereum-mainnet-zircuit-1', mainnet_evm_ethereum_mainnet_zircuit_1],
        ['memento-mainnet', mainnet_evm_memento_mainnet],
        ['ethereum-mainnet-ink-1', mainnet_evm_ethereum_mainnet_ink_1],
        ['ethereum-mainnet-linea-1', mainnet_evm_ethereum_mainnet_linea_1],
        ['nexon-mainnet-lith', mainnet_evm_nexon_mainnet_lith],
        ['bitcoin-mainnet-bob-1', mainnet_evm_bitcoin_mainnet_bob_1],
        ['treasure-mainnet', mainnet_evm_treasure_mainnet],
        ['nexon-mainnet-henesys', mainnet_evm_nexon_mainnet_henesys],
        ['berachain-mainnet', mainnet_evm_berachain_mainnet],
        ['codex-mainnet', mainnet_evm_codex_mainnet],
        ['ethereum-mainnet-blast-1', mainnet_evm_ethereum_mainnet_blast_1],
        ['plume-mainnet', mainnet_evm_plume_mainnet],
        ['ethereum-mainnet-taiko-1', mainnet_evm_ethereum_mainnet_taiko_1],
        ['bitcoin-mainnet-bitlayer-1', mainnet_evm_bitcoin_mainnet_bitlayer_1],
        ['avalanche-subnet-dexalot-mainnet', mainnet_evm_avalanche_subnet_dexalot_mainnet],
        ['ethereum-mainnet-scroll-1', mainnet_evm_ethereum_mainnet_scroll_1],
        ['polygon-mainnet-katana', mainnet_evm_polygon_mainnet_katana],
        ['nexon-qa', mainnet_evm_nexon_qa],
        ['zklink_nova-mainnet', mainnet_evm_zklink_nova_mainnet],
        ['nexon-stage', mainnet_evm_nexon_stage],
        ['ethereum-mainnet-arbitrum-1-treasure-1', mainnet_evm_ethereum_mainnet_arbitrum_1_treasure_1],
        ['zora-mainnet', mainnet_evm_zora_mainnet],
        ['corn-mainnet', mainnet_evm_corn_mainnet],
        ['tron-mainnet-evm', mainnet_evm_tron_mainnet_evm],
    ]),
    solana: new Map([['solana-mainnet', mainnet_solana_solana_mainnet]]),
    aptos: new Map([['aptos-mainnet', mainnet_aptos_aptos_mainnet]]),
    sui: new Map([['sui-mainnet', mainnet_sui_sui_mainnet]]),
    ton: new Map([['ton-mainnet', mainnet_ton_ton_mainnet]]),
    tron: new Map([['tron-mainnet', mainnet_tron_tron_mainnet]]),
};
export const testnetByNameByFamily = {
    evm: new Map([
        ['bitcoin-testnet-rootstock', testnet_evm_bitcoin_testnet_rootstock],
        ['telos-evm-testnet', testnet_evm_telos_evm_testnet],
        ['polkadot-testnet-darwinia-pangoro', testnet_evm_polkadot_testnet_darwinia_pangoro],
        ['xdc-testnet', testnet_evm_xdc_testnet],
        ['coinex_smart_chain-testnet', testnet_evm_coinex_smart_chain_testnet],
        ['polkadot-testnet-astar-shibuya', testnet_evm_polkadot_testnet_astar_shibuya],
        ['binance_smart_chain-testnet', testnet_evm_binance_smart_chain_testnet],
        ['velas-testnet', testnet_evm_velas_testnet],
        ['ethereum-testnet-sepolia-hashkey-1', testnet_evm_ethereum_testnet_sepolia_hashkey_1],
        ['shibarium-testnet-puppynet', testnet_evm_shibarium_testnet_puppynet],
        ['ethereum-testnet-sepolia-xlayer-1', testnet_evm_ethereum_testnet_sepolia_xlayer_1],
        ['cronos-zkevm-testnet-sepolia', testnet_evm_cronos_zkevm_testnet_sepolia],
        ['ethereum-testnet-goerli-zksync-1', testnet_evm_ethereum_testnet_goerli_zksync_1],
        ['cronos-testnet-zkevm-1', testnet_evm_cronos_testnet_zkevm_1],
        ['hedera-testnet', testnet_evm_hedera_testnet],
        ['ethereum-testnet-sepolia-zksync-1', testnet_evm_ethereum_testnet_sepolia_zksync_1],
        ['cronos-testnet', testnet_evm_cronos_testnet],
        ['near-testnet', testnet_evm_near_testnet],
        ['ethereum-testnet-goerli-optimism-1', testnet_evm_ethereum_testnet_goerli_optimism_1],
        ['areon-testnet', testnet_evm_areon_testnet],
        ['janction-testnet-sepolia', testnet_evm_janction_testnet_sepolia],
        ['private-testnet-obsidian', testnet_evm_private_testnet_obsidian],
        ['ethereum-testnet-sepolia-mode-1', testnet_evm_ethereum_testnet_sepolia_mode_1],
        ['bittensor-testnet', testnet_evm_bittensor_testnet],
        ['hyperliquid-testnet', testnet_evm_hyperliquid_testnet],
        ['kaia-testnet-kairos', testnet_evm_kaia_testnet_kairos],
        ['bittorrent_chain-testnet', testnet_evm_bittorrent_chain_testnet],
        ['wemix-testnet', testnet_evm_wemix_testnet],
        ['core-testnet', testnet_evm_core_testnet],
        ['bitcoin-testnet-bsquared-1', testnet_evm_bitcoin_testnet_bsquared_1],
        ['polkadot-testnet-moonbeam-moonbase', testnet_evm_polkadot_testnet_moonbeam_moonbase],
        ['ethereum-testnet-sepolia-unichain-1', testnet_evm_ethereum_testnet_sepolia_unichain_1],
        ['sei-testnet-atlantic', testnet_evm_sei_testnet_atlantic],
        ['geth-testnet', testnet_evm_geth_testnet],
        [
            'ethereum-testnet-goerli-polygon-zkevm-1',
            testnet_evm_ethereum_testnet_goerli_polygon_zkevm_1,
        ],
        ['story-testnet', testnet_evm_story_testnet],
        ['mint-testnet', testnet_evm_mint_testnet],
        ['metal-testnet', testnet_evm_metal_testnet],
        ['bitcichain-testnet', testnet_evm_bitcichain_testnet],
        ['ethereum-testnet-sepolia-soneium-1', testnet_evm_ethereum_testnet_sepolia_soneium_1],
        ['ronin-testnet-saigon', testnet_evm_ronin_testnet_saigon],
        ['private-testnet-granite', testnet_evm_private_testnet_granite],
        ['private-testnet-andesite', testnet_evm_private_testnet_andesite],
        ['dtcc-testnet-andesite', testnet_evm_dtcc_testnet_andesite],
        ['polkadot-testnet-centrifuge-altair', testnet_evm_polkadot_testnet_centrifuge_altair],
        ['memento-testnet', testnet_evm_memento_testnet],
        ['kava-testnet', testnet_evm_kava_testnet],
        ['ethereum-testnet-sepolia-kroma-1', testnet_evm_ethereum_testnet_sepolia_kroma_1],
        ['tac-testnet', testnet_evm_tac_testnet],
        [
            'ethereum-testnet-sepolia-polygon-zkevm-1',
            testnet_evm_ethereum_testnet_sepolia_polygon_zkevm_1,
        ],
        ['ethereum-testnet-holesky-fraxtal-1', testnet_evm_ethereum_testnet_holesky_fraxtal_1],
        ['ethereum-testnet-holesky-morph-1', testnet_evm_ethereum_testnet_holesky_morph_1],
        ['bitcoin-testnet-botanix', testnet_evm_bitcoin_testnet_botanix],
        ['fantom-testnet', testnet_evm_fantom_testnet],
        ['ethereum-testnet-sepolia-lisk-1', testnet_evm_ethereum_testnet_sepolia_lisk_1],
        ['ethereum-testnet-sepolia-worldchain-1', testnet_evm_ethereum_testnet_sepolia_worldchain_1],
        ['ethereum-testnet-goerli-mantle-1', testnet_evm_ethereum_testnet_goerli_mantle_1],
        ['ethereum-testnet-sepolia-mantle-1', testnet_evm_ethereum_testnet_sepolia_mantle_1],
        ['binance_smart_chain-testnet-opbnb-1', testnet_evm_binance_smart_chain_testnet_opbnb_1],
        ['nexon-dev', testnet_evm_nexon_dev],
        ['megaeth-testnet', testnet_evm_megaeth_testnet],
        ['nibiru-testnet', testnet_evm_nibiru_testnet],
        ['ondo-testnet', testnet_evm_ondo_testnet],
        ['neonlink-testnet', testnet_evm_neonlink_testnet],
        ['plasma-testnet', testnet_evm_plasma_testnet],
        ['monad-testnet', testnet_evm_monad_testnet],
        ['gnosis_chain-testnet-chiado', testnet_evm_gnosis_chain_testnet_chiado],
        ['abstract-testnet', testnet_evm_abstract_testnet],
        [
            'ethereum-testnet-sepolia-arbitrum-1-l3x-1',
            testnet_evm_ethereum_testnet_sepolia_arbitrum_1_l3x_1,
        ],
        [
            'ethereum-testnet-sepolia-immutable-zkevm-1',
            testnet_evm_ethereum_testnet_sepolia_immutable_zkevm_1,
        ],
        ['0g-testnet-newton', testnet_evm_0g_testnet_newton],
        ['0g-testnet-galileo', testnet_evm_0g_testnet_galileo],
        ['ethereum-testnet-holesky', testnet_evm_ethereum_testnet_holesky],
        ['anvil-devnet', testnet_evm_anvil_devnet],
        ['apechain-testnet-curtis', testnet_evm_apechain_testnet_curtis],
        ['ethereum-testnet-sepolia-lens-1', testnet_evm_ethereum_testnet_sepolia_lens_1],
        ['avalanche-testnet-fuji', testnet_evm_avalanche_testnet_fuji],
        ['celo-testnet-alfajores', testnet_evm_celo_testnet_alfajores],
        ['private-testnet-opala', testnet_evm_private_testnet_opala],
        ['zircuit-testnet-garfield', testnet_evm_zircuit_testnet_garfield],
        ['ethereum-testnet-sepolia-zircuit-1', testnet_evm_ethereum_testnet_sepolia_zircuit_1],
        ['superseed-testnet', testnet_evm_superseed_testnet],
        ['sonic-testnet-blaze', testnet_evm_sonic_testnet_blaze],
        ['ethereum-testnet-goerli-linea-1', testnet_evm_ethereum_testnet_goerli_linea_1],
        ['ethereum-testnet-sepolia-linea-1', testnet_evm_ethereum_testnet_sepolia_linea_1],
        ['ethereum-testnet-sepolia-metis-1', testnet_evm_ethereum_testnet_sepolia_metis_1],
        ['polygon-testnet-mumbai', testnet_evm_polygon_testnet_mumbai],
        ['polygon-testnet-amoy', testnet_evm_polygon_testnet_amoy],
        ['berachain-testnet-bepolia', testnet_evm_berachain_testnet_bepolia],
        ['berachain-testnet-bartio', testnet_evm_berachain_testnet_bartio],
        ['berachain-testnet-artio', testnet_evm_berachain_testnet_artio],
        ['zero-g-testnet-galileo', testnet_evm_zero_g_testnet_galileo],
        ['ethereum-testnet-goerli-base-1', testnet_evm_ethereum_testnet_goerli_base_1],
        ['ethereum-testnet-sepolia-base-1', testnet_evm_ethereum_testnet_sepolia_base_1],
        ['plume-devnet', testnet_evm_plume_devnet],
        ['plume-testnet-sepolia', testnet_evm_plume_testnet_sepolia],
        ['etherlink-testnet', testnet_evm_etherlink_testnet],
        ['polygon-testnet-tatara', testnet_evm_polygon_testnet_tatara],
        ['ethereum-testnet-holesky-taiko-1', testnet_evm_ethereum_testnet_holesky_taiko_1],
        ['mind-testnet', testnet_evm_mind_testnet],
        ['bitcoin-testnet-bitlayer-1', testnet_evm_bitcoin_testnet_bitlayer_1],
        ['ethereum-testnet-goerli-arbitrum-1', testnet_evm_ethereum_testnet_goerli_arbitrum_1],
        ['ethereum-testnet-sepolia-arbitrum-1', testnet_evm_ethereum_testnet_sepolia_arbitrum_1],
        ['private-testnet-mica', testnet_evm_private_testnet_mica],
        ['avalanche-subnet-dexalot-testnet', testnet_evm_avalanche_subnet_dexalot_testnet],
        ['ethereum-testnet-sepolia-scroll-1', testnet_evm_ethereum_testnet_sepolia_scroll_1],
        ['avalanche-testnet-nexon', testnet_evm_avalanche_testnet_nexon],
        ['bitcoin-testnet-merlin', testnet_evm_bitcoin_testnet_merlin],
        ['pharos-testnet', testnet_evm_pharos_testnet],
        [
            'ethereum-testnet-sepolia-polygon-validium-1',
            testnet_evm_ethereum_testnet_sepolia_polygon_validium_1,
        ],
        ['hemi-testnet-sepolia', testnet_evm_hemi_testnet_sepolia],
        ['ink-testnet-sepolia', testnet_evm_ink_testnet_sepolia],
        ['bitcoin-testnet-sepolia-bob-1', testnet_evm_bitcoin_testnet_sepolia_bob_1],
        ['zklink_nova-testnet', testnet_evm_zklink_nova_testnet],
        ['codex-testnet', testnet_evm_codex_testnet],
        [
            'ethereum-testnet-sepolia-arbitrum-1-treasure-1',
            testnet_evm_ethereum_testnet_sepolia_arbitrum_1_treasure_1,
        ],
        ['treasure-testnet-topaz', testnet_evm_treasure_testnet_topaz],
        ['jovay-testnet', testnet_evm_jovay_testnet],
        ['ethereum-testnet-sepolia', testnet_evm_ethereum_testnet_sepolia],
        ['ethereum-testnet-sepolia-optimism-1', testnet_evm_ethereum_testnet_sepolia_optimism_1],
        ['neox-testnet-t4', testnet_evm_neox_testnet_t4],
        ['ethereum-testnet-sepolia-corn-1', testnet_evm_ethereum_testnet_sepolia_corn_1],
        ['filecoin-testnet', testnet_evm_filecoin_testnet],
        ['plume-testnet', testnet_evm_plume_testnet],
        ['ethereum-testnet-sepolia-blast-1', testnet_evm_ethereum_testnet_sepolia_blast_1],
        ['zora-testnet', testnet_evm_zora_testnet],
        ['tron-testnet-shasta-evm', testnet_evm_tron_testnet_shasta_evm],
        ['tron-devnet-evm', testnet_evm_tron_devnet_evm],
        ['tron-testnet-nile-evm', testnet_evm_tron_testnet_nile_evm],
    ]),
    solana: new Map([
        ['solana-testnet', testnet_solana_solana_testnet],
        ['solana-devnet', testnet_solana_solana_devnet],
    ]),
    aptos: new Map([
        ['aptos-testnet', testnet_aptos_aptos_testnet],
        ['aptos-localnet', testnet_aptos_aptos_localnet],
    ]),
    sui: new Map([
        ['sui-testnet', testnet_sui_sui_testnet],
        ['sui-localnet', testnet_sui_sui_localnet],
    ]),
    ton: new Map([
        ['ton-testnet', testnet_ton_ton_testnet],
        ['ton-localnet', testnet_ton_ton_localnet],
    ]),
    tron: new Map([
        ['tron-testnet-shasta', testnet_tron_tron_testnet_shasta],
        ['tron-devnet', testnet_tron_tron_devnet],
        ['tron-testnet-nile', testnet_tron_tron_testnet_nile],
    ]),
};
