// @chainlink/cre-sdk/dist/sdk/utils/capabilities/http/http-helpers.js
import { decodeJson } from '../../decode-json';
export function text(responseOrFn) {
    if (typeof responseOrFn === 'function') {
        return {
            result: () => text(responseOrFn().result),
        };
    }
    else {
        const decoder = new TextDecoder('utf-8');
        return decoder.decode(responseOrFn.body).trim();
    }
}
export function json(responseOrFn) {
    if (typeof responseOrFn === 'function') {
        return {
            result: () => json(responseOrFn().result),
        };
    }
    return decodeJson(responseOrFn.body);
}
export function getHeader(responseOrFn, name) {
    if (typeof responseOrFn === 'function') {
        return {
            result: () => getHeader(responseOrFn().result, name),
        };
    }
    else {
        const lowerName = name.toLowerCase();
        return Object.entries(responseOrFn.headers).find(([key]) => key.toLowerCase() === lowerName)?.[1];
    }
}
export function ok(responseOrFn) {
    if (typeof responseOrFn === 'function') {
        return {
            result: () => ok(responseOrFn().result),
        };
    }
    else {
        return responseOrFn.statusCode >= 200 && responseOrFn.statusCode < 300;
    }
}
// ============================================================================
// SendReport Helper Methods for ClientCapability and SendRequester
// ============================================================================
/**
 * SendReport functions the same as SendRequest, but takes a Report and a function
 * to convert the inner ReportResponse to a Request.
 * Note that caching is limited as reports may contain different sets of signatures
 * on different nodes, leading to a cache miss.
 *
 * @param runtime - The runtime instance
 * @param report - The Report to process
 * @param fn - Function to convert ReportResponse to Request
 * @returns Response result function
 */
function sendReport(runtime, report, fn) {
    const rawReport = report.x_generatedCodeOnly_unwrap();
    const request = fn(rawReport);
    return this.sendRequest(runtime, request);
}
/**
 * SendReport functions the same as SendRequest, but takes a Report and a function
 * to convert the inner ReportResponse to a Request.
 * Note that caching is limited as reports may contain different sets of signatures
 * on different nodes, leading to a cache miss.
 *
 * @param report - The Report to process
 * @param fn - Function to convert ReportResponse to Request
 * @returns Response result function
 */
function sendRequesterSendReport(report, fn) {
    const rawReport = report.x_generatedCodeOnly_unwrap();
    const request = fn(rawReport);
    return this.sendRequest(request);
}
// ============================================================================
// Prototype Extensions
// ============================================================================
// Import the actual classes for prototype extension
import { ClientCapability as ClientCapabilityClass, SendRequester as SendRequesterClass, } from '../../../../generated-sdk/capabilities/networking/http/v1alpha/client_sdk_gen';
// Extend ClientCapability prototype
ClientCapabilityClass.prototype.sendReport = sendReport;
// Extend SendRequester prototype
SendRequesterClass.prototype.sendReport = sendRequesterSendReport;
