// @chainlink/cre-sdk/dist/generated-sdk/capabilities/scheduler/cron/v1/cron_sdk_gen.js
import { create, fromJson } from '@bufbuild/protobuf';
import { AnySchema, anyPack } from '@bufbuild/protobuf/wkt';
import { ConfigSchema, LegacyPayloadSchema, PayloadSchema, } from '../../../../../generated/capabilities/scheduler/cron/v1/trigger_pb';
/**
 * Cron Capability
 *
 * Capability ID: cron-trigger@1.0.0
 * Capability Name: cron-trigger
 * Capability Version: 1.0.0
 */
export class CronCapability {
    /** The capability ID for this service */
    static CAPABILITY_ID = 'cron-trigger@1.0.0';
    static CAPABILITY_NAME = 'cron-trigger';
    static CAPABILITY_VERSION = '1.0.0';
    trigger(config) {
        const capabilityId = CronCapability.CAPABILITY_ID;
        return new CronTrigger(config, capabilityId, 'Trigger');
    }
}
/**
 * Trigger implementation for Trigger
 */
class CronTrigger {
    _capabilityId;
    _method;
    config;
    constructor(config, _capabilityId, _method) {
        this._capabilityId = _capabilityId;
        this._method = _method;
        // biome-ignore lint/suspicious/noExplicitAny: Needed for runtime type checking of protocol buffer messages
        this.config = config.$typeName
            ? config
            : fromJson(ConfigSchema, config);
    }
    capabilityId() {
        return this._capabilityId;
    }
    method() {
        return this._method;
    }
    outputSchema() {
        return PayloadSchema;
    }
    configAsAny() {
        return anyPack(ConfigSchema, this.config);
    }
    /**
     * Transform the raw trigger output - override this method if needed
     * Default implementation returns the raw output unchanged
     */
    adapt(rawOutput) {
        return rawOutput;
    }
}
