// @chainlink/cre-sdk/dist/generated-sdk/capabilities/internal/consensus/v1alpha/consensus_sdk_gen.js
import { fromJson } from '@bufbuild/protobuf';
import { ReportRequestSchema, ReportResponseSchema, SimpleConsensusInputsSchema, } from '../../../../../generated/sdk/v1alpha/sdk_pb';
import { ValueSchema } from '../../../../../generated/values/v1/values_pb';
import { Report } from '../../../../../sdk/report';
import { hexToBytes } from '../../../../../sdk/utils/hex-utils';
/**
 * Consensus Capability
 *
 * Capability ID: consensus@1.0.0-alpha
 * Capability Name: consensus
 * Capability Version: 1.0.0-alpha
 */
export class ConsensusCapability {
    /** The capability ID for this service */
    static CAPABILITY_ID = 'consensus@1.0.0-alpha';
    static CAPABILITY_NAME = 'consensus';
    static CAPABILITY_VERSION = '1.0.0-alpha';
    simple(runtime, input) {
        // Handle input conversion - unwrap if it's a wrapped type, convert from JSON if needed
        let payload;
        if (input.$typeName) {
            // It's the original protobuf type
            payload = input;
        }
        else {
            // It's regular JSON, convert using fromJson
            payload = fromJson(SimpleConsensusInputsSchema, input);
        }
        const capabilityId = ConsensusCapability.CAPABILITY_ID;
        const capabilityResponse = runtime.callCapability({
            capabilityId,
            method: 'Simple',
            payload,
            inputSchema: SimpleConsensusInputsSchema,
            outputSchema: ValueSchema,
        });
        return {
            result: () => {
                const result = capabilityResponse.result();
                return result;
            },
        };
    }
    report(runtime, input) {
        // Handle input conversion - unwrap if it's a wrapped type, convert from JSON if needed
        let payload;
        if (input.$typeName) {
            // It's the original protobuf type
            payload = input;
        }
        else {
            // It's regular JSON, convert using fromJson
            payload = fromJson(ReportRequestSchema, input);
        }
        const capabilityId = ConsensusCapability.CAPABILITY_ID;
        const capabilityResponse = runtime.callCapability({
            capabilityId,
            method: 'Report',
            payload,
            inputSchema: ReportRequestSchema,
            outputSchema: ReportResponseSchema,
        });
        return {
            result: () => {
                const result = capabilityResponse.result();
                return new Report(result);
            },
        };
    }
}
