// @chainlink/cre-sdk/dist/generated-sdk/capabilities/networking/confidentialhttp/v1alpha/client_sdk_gen.js
import { fromJson } from '@bufbuild/protobuf';
import { ConfidentialHTTPRequestSchema, HTTPResponseSchema, } from '../../../../../generated/capabilities/networking/confidentialhttp/v1alpha/client_pb';
import { Report } from '../../../../../sdk/report';
import { hexToBytes } from '../../../../../sdk/utils/hex-utils';
/**
 * Client Capability
 *
 * Capability ID: confidential-http@1.0.0-alpha
 * Capability Name: confidential-http
 * Capability Version: 1.0.0-alpha
 */
export class ClientCapability {
    /** The capability ID for this service */
    static CAPABILITY_ID = 'confidential-http@1.0.0-alpha';
    static CAPABILITY_NAME = 'confidential-http';
    static CAPABILITY_VERSION = '1.0.0-alpha';
    sendRequest(runtime, input) {
        // Handle input conversion - unwrap if it's a wrapped type, convert from JSON if needed
        let payload;
        if (input.$typeName) {
            // It's the original protobuf type
            payload = input;
        }
        else {
            // It's regular JSON, convert using fromJson
            payload = fromJson(ConfidentialHTTPRequestSchema, input);
        }
        const capabilityId = ClientCapability.CAPABILITY_ID;
        const capabilityResponse = runtime.callCapability({
            capabilityId,
            method: 'SendRequest',
            payload,
            inputSchema: ConfidentialHTTPRequestSchema,
            outputSchema: HTTPResponseSchema,
        });
        return {
            result: () => {
                const result = capabilityResponse.result();
                return result;
            },
        };
    }
}
