// @chainlink/cre-sdk/dist/sdk/wasm/host-bindings.js
import { Mode } from '../../generated/sdk/v1alpha/sdk_pb';
import { z } from 'zod';
// Zod schema for validating global host functions
const globalHostBindingsSchema = z.object({
    switchModes: z.function().args(z.nativeEnum(Mode)).returns(z.void()),
    log: z.function().args(z.string()).returns(z.void()),
    sendResponse: z
        .function()
        .args(z.union([z.instanceof(Uint8Array), z.custom()]))
        .returns(z.number()),
    versionV2: z.function().args().returns(z.void()),
    callCapability: z
        .function()
        .args(z.union([z.instanceof(Uint8Array), z.custom()]))
        .returns(z.number()),
    awaitCapabilities: z
        .function()
        .args(z.union([z.instanceof(Uint8Array), z.custom()]), z.number())
        .returns(z.union([z.instanceof(Uint8Array), z.custom()])),
    getSecrets: z
        .function()
        .args(z.union([z.instanceof(Uint8Array), z.custom()]), z.number())
        .returns(z.any()),
    awaitSecrets: z
        .function()
        .args(z.union([z.instanceof(Uint8Array), z.custom()]), z.number())
        .returns(z.union([z.instanceof(Uint8Array), z.custom()])),
    getWasiArgs: z.function().args().returns(z.string()),
    now: z.function().args().returns(z.number()),
});
// Validate global host functions at runtime
const validateGlobalHostBindings = () => {
    const globalFunctions = globalThis;
    try {
        return globalHostBindingsSchema.parse(globalFunctions);
    }
    catch (error) {
        const missingFunctions = Object.keys(globalHostBindingsSchema.shape).filter((key) => !(key in globalFunctions));
        throw new Error(`Missing required global host functions: ${missingFunctions.join(', ')}. ` +
            `The CRE WASM runtime must provide these functions on globalThis. ` +
            `This usually means the workflow is being executed outside the CRE WASM environment, ` +
            `or the host runtime version is incompatible with this SDK version.`);
    }
};
// Initialize validated global functions (lazy evaluation for testing)
let _hostBindings = null;
export const hostBindings = new Proxy({}, {
    get(target, prop) {
        if (!_hostBindings) {
            _hostBindings = validateGlobalHostBindings();
        }
        return _hostBindings[prop];
    },
});
