// @chainlink/cre-sdk/dist/sdk/wasm/runtime.js
import { fromBinary, toBinary } from '@bufbuild/protobuf';
import { AwaitCapabilitiesRequestSchema, AwaitCapabilitiesResponseSchema, AwaitSecretsRequestSchema, AwaitSecretsResponseSchema, CapabilityRequestSchema, GetSecretsRequestSchema, } from '../../generated/sdk/v1alpha/sdk_pb';
import { NodeRuntimeImpl, RuntimeImpl } from '../impl/runtime-impl';
import { hostBindings } from './host-bindings';
export class Runtime extends RuntimeImpl {
    constructor(config, nextCallId, maxResponseSize) {
        super(config, nextCallId, WasmRuntimeHelpers.getInstance(), maxResponseSize);
    }
}
export class NodeRuntime extends NodeRuntimeImpl {
    constructor(config, nextCallId, maxResponseSize) {
        super(config, nextCallId, WasmRuntimeHelpers.getInstance(), maxResponseSize);
    }
}
/** Convert bigint maxResponseSize to i32 for WASM host binding, with range validation. */
function toI32ResponseSize(maxResponseSize) {
    if (maxResponseSize > 2147483647n || maxResponseSize < -2147483648n) {
        throw new Error(`maxResponseSize ${maxResponseSize} exceeds i32 range. Expected a value between -2147483648 and 2147483647`);
    }
    return Math.trunc(Number(maxResponseSize));
}
class WasmRuntimeHelpers {
    static instance;
    constructor() { }
    now() {
        return hostBindings.now();
    }
    static getInstance() {
        if (!WasmRuntimeHelpers.instance) {
            WasmRuntimeHelpers.instance = new WasmRuntimeHelpers();
        }
        return WasmRuntimeHelpers.instance;
    }
    call(request) {
        return hostBindings.callCapability(toBinary(CapabilityRequestSchema, request)) >= 0;
    }
    await(request, maxResponseSize) {
        const responseSize = toI32ResponseSize(maxResponseSize);
        const response = hostBindings.awaitCapabilities(toBinary(AwaitCapabilitiesRequestSchema, request), responseSize);
        const responseBytes = Array.isArray(response) ? new Uint8Array(response) : response;
        return fromBinary(AwaitCapabilitiesResponseSchema, responseBytes);
    }
    getSecrets(request, maxResponseSize) {
        const responseSize = toI32ResponseSize(maxResponseSize);
        return hostBindings.getSecrets(toBinary(GetSecretsRequestSchema, request), responseSize) >= 0;
    }
    awaitSecrets(request, maxResponseSize) {
        const responseSize = toI32ResponseSize(maxResponseSize);
        const response = hostBindings.awaitSecrets(toBinary(AwaitSecretsRequestSchema, request), responseSize);
        const responseBytes = Array.isArray(response) ? new Uint8Array(response) : response;
        return fromBinary(AwaitSecretsResponseSchema, responseBytes);
    }
    switchModes(mode) {
        hostBindings.switchModes(mode);
    }
    log(message) {
        hostBindings.log(message);
    }
}
